---
title: "CHIR HIV RV306 Vaginal - Populate mre"
author: "Francesc Català-Moll & Oriol Careta Borràs"
output: html_document
output_dir: ../docs
site: workflowr::wflow_site
mainfont: Helvetica
date: "`r format(Sys.time(), '%Y-%m-%d')`"
header-includes:
- \usepackage{fancyhdr}
- \usepackage{lipsum}
- \pagestyle{fancy}
- \fancyhead[CO,CE]{Flamingo - CIHR}
- \fancyfoot[LO,RE]{Oriol Careta Borras}
- \fancyhead[LE,RO]{\thepage}
- \fancyfoot[CO,CE]{}
- \usepackage[default]{sourcesanspro}
- \usepackage[T1]{fontenc}
- \usepackage{eso-pic,graphicx,transparent}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  cache = FALSE,
  warning = FALSE,
  message = FALSE,
  dpi = 180,
  fig.align = "center",
  comment = "#>",
  fig.width = 20,
  fig.height = 10
)
knitr::opts_knit$set(root.dir = here::here())
```

## Introduction

The initial MRE object, containing empty result slots, will be utilized to run a variety of statistical tests using project-specific categorical and numerical variables. Pre-programmed functions will populate the slots with results generated from the following analyses: 

*ordination
*differential abundance
*correlation
*gene richness
*biomarker discovery
*gene function
*other tests that can be programmed from scratch or built on pre-run statistical testing.

Each analysis will produce a set of result files, usually in the form of PDF or CSV files, which can be used for exploration and further analysis. Once all tests have been completed, the MRE object will be populated with the corresponding results and saved to disk or AWS for future reference.

## Load mre

```{r load_mre}
## mre init
mre <- readr::read_rds(here::here("data", "preprocessed", "mre.rds"))
```

## Populate diversity

This section focuses on analyzing alpha diversity, which refers to the diversity within a particular sample. It includes two types of analyses:

**Linear mixed model**: This analysis is used to analyze longitudinal or clustered data and it explores the association between microbial diversity (alpha diversity) and metadata variables using a linear mixed model approach. It helps understand how different factors influence the diversity of microbial communities within samples.

```{r populate_diversity_lmm}
library(metar)
mre <- virgo_lmm(mre, save_files = FALSE)
```

**Rarefaction**: Rarefaction curves are used to estimate species richness in a sample. It involves randomly subsampling sequences to estimate the number of different species present, which helps in assessing whether sampling depth is sufficient to capture the microbial diversity present in the samples.

```{r}
mre <- virgo_rarefaction(mre, save_files = FALSE)
```

## Populate taxa

This section involves downstream taxonomic analysis of microbiome data, which focuses on identifying and characterizing microbial taxa present in the samples. It includes various analyses:

**Analysis of composition of microbiomes (AnCoM)**: AnCoM is a statistical framework specifically designed for compositional data analysis used to identify taxa that differ significantly in abundance between groups defined by categorical metadata variables.

```{r populate_taxa_ancom}
mre <- virgo_ancom(mre, save_files = FALSE)
```

**Barplots ordered using non-Metric Multidimensional Scaling (NMDS) ordering**: NMDS is a dimensionality reduction technique used to visualize the similarity/dissimilarity of microbial communities across samples based on taxonomic composition. Barplots are used to illustrate the relative abundances of different taxa across samples, providing insights into community structure.

```{r populate_taxa_barplots}
mre <- virgo_barplots(mre, save_files = FALSE)
```

**Differential gene expression analysis based on the negative binomial distribution**: Used to identify taxa exhibiting significant abundance differences between conditions while accounting for overdispersion.

```{r populate_taxa_deseq}
mre <- virgo_deseq(mre, save_files = FALSE)
```

**Hierarchical Clustering Analysis**: This analysis groups samples based on their taxonomic composition similarity, providing insights into the overall structure of microbial communities. It helps to visualize patterns of taxonomic abundance across samples.

```{r populate_taxa_heatmap}
mre <- virgo_heatmap(mre, save_files = FALSE)
```

**Microbiome Multivariable Association with Linear Models (MAAsLin)**: This method explores associations between microbial taxa and multiple metadata variables simultaneously using linear models while adjusting for confounding factors.

```{r populate_taxa_maaslin}
mre <- virgo_maaslin(mre, save_files = FALSE)
```

**Statistical analysis for sparse high-throughput sequencing**: This method is used for analyzing metagenomic sequencing data, providing methods to identify taxa with significantly different abundances between conditions while addressing issues such as overdispersion and zero-inflation.

```{r populate_taxa_metagenomeseq}
mre <- virgo_metagenomeSeq(mre, save_files = FALSE)
```

**Partial least squares discriminant analysis (PLS-DA)**: PLS-DA is a supervised dimensionality reduction technique used to find the variables that explain the maximum variance in data while also discriminating between predefined groups.

```{r populate_taxa_plsda}
mre <- virgo_plsda(mre, save_files = FALSE)
```

**Forward-selection method for the identification of two groups of taxa whose relative abundance, or balance, is associated with the response variable of interest**: This method is designed to identify associations between microbial taxa and metadata while considering the sparse nature of microbiome data, potentially uncovering subtle yet meaningful relationships.

```{r populate_taxa_sebal}
mre <- virgo_selbal(mre, save_files = FALSE)
```

**Statistical Inference of Associations between Microbial Communities And host phenoTypes **: It performs statistical modelling using LASSO logistic regression to determine changes in community composition that are associated with environmental factors, focusing on association with host phenotypes.

```{r populate_taxa_siamcat}
mre <- virgo_siamcat(mre, save_files = FALSE)
```

**ANOVA-Like Differential Expression**: This analysis identifies differentially abundant taxa across experimental groups using an ANOVA-like approach, considering the compositional nature of microbial abundances and addressing issues related to sampling variability.

```{r populate_taxa_aldex}
mre <- virgo_aldex(mre, save_files = FALSE)
```

**Count Regression for Correlated Observations with the Beta-binomial**: This method models count data with overdispersion, which is common in microbiome data, using a beta-binomial distribution.

```{r populate_taxa_cornob}
mre <- virgo_corncob(mre, save_files = FALSE)
```

**Linear discriminant analysis Effect Size (LEfSe)**: LEfSe identifies taxa characterizing the differences between two or more biological conditions, often used to find biomarkers associated with specific conditions.

```{r populate_taxa_lefse}
mre <- virgo_lefse(mre, save_files = FALSE)
```

**Ordination Analysis using Non-metric multidimensional scaling (NMDS)**: Similar to the earlier NMDS analysis, this likely visualizes taxonomic composition data in lower-dimensional space to observe patterns and associations.

```{r populate_taxa_nmds}
mre <- virgo_nmds(mre, save_files = FALSE)
```

## Populate functional

This section focuses on functional analysis, which involves characterizing the functional capabilities of microbial communities. It includes:

**Normalization and group-wise comparisons**: This involves performing functional analysis by associating differentially expressed genes with known functions from the KEGG database.

```{r functional_deseq}
mre <- functional_deseq(mre, save_files = FALSE, type = "virgo", db = "kegg")
```

**Functional heatmap**: This visualization method likely shows the abundance of functional gene categories across samples, providing insights into functional differences between groups.

```{r functional_heatmap}
mre <- functional_heatmap(mre, save_files = FALSE, type = "virgo", db = "kegg") 
```

**Functional maaslin**: This visualization method likely shows a differential functional annotation analysis with Maaslin.

```{r functional_maaslin}
mre <- functional_maaslin(mre, save_files = FALSE, type = "virgo", db = "kegg") 
```

**Functional nmnds**: NMDS analysis wrapper for functional data.

```{r functional_nmnds}
phy <- 
  metar::get_phyloseq(mre, type = "virgo") %>%
  microbiome::transform(transform = "compositional") %>%
  microbiome::core(detection = 0.001, prevalence = 0.1)
mre <- functional_nmnds(mre, save_files = FALSE, type = "virgo", db = "kegg") 
```

## Save mre

```{r save_mre}
out_dir <- here::here("data/preprocessed/")
readr::write_rds(mre, file = here::here(out_dir, "mre.rds"))
```

