<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Francesc Català-Moll &amp; Oriol Careta Borràs" />

<meta name="date" content="2024-04-19" />

<title>CHIR HIV RV306 Vaginal - Throughput Analysis</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">2024_cihr_hiv_rv306</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="004_throughput_analysis.html">Throughput Analysis</a>
</li>
<li>
  <a href="005_mre_initialization.html">Init mre object</a>
</li>
<li>
  <a href="007_phylosmith_abundance.html">Phylosmith abundance exploration</a>
</li>
<li>
  <a href="008_phylosmith_diversity.html">Phylosmith diversity exploration</a>
</li>
<li>
  <a href="009_quality_control.html">Quality_control</a>
</li>
<li>
  <a href="100_global_description.html">Global description</a>
</li>
<li>
  <a href="101_baseline_timepoints.html">Baseline timepoints</a>
</li>
<li>
  <a href="102_starting_point_description.html">Starting point description</a>
</li>
<li>
  <a href="103_baseline_description.html">Baseline description</a>
</li>
<li>
  <a href="104_groups_comparison.html">Groups Comparison</a>
</li>
<li>
  <a href="105_longitudinal_sudy_t1.html">Longitudinal Study of Group 1</a>
</li>
<li>
  <a href="105_longitudinal_sudy_t2.html">Longitudinal Study of Group 2</a>
</li>
<li>
  <a href="105_longitudinal_sudy_t3.html">Longitudinal Study of Group 3</a>
</li>
<li>
  <a href="105_longitudinal_sudy_t4A.html">Longitudinal Study of Group 4A</a>
</li>
<li>
  <a href="105_longitudinal_sudy_t4B.html">Longitudinal Study of Group 4B</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">CHIR HIV RV306 Vaginal - Throughput
Analysis</h1>
<h4 class="author">Francesc Català-Moll &amp; Oriol Careta Borràs</h4>
<h4 class="date">2024-04-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-04-19
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>2024_chir_hiv_rv306_vaginal/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240321code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240321)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240321code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240321)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong412fedd">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 412fedd
</a>
</p>
</div>
<div id="strongRepositoryversionstrong412fedd"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 412fedd.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/Study_1.png
    Ignored:    analysis/Study_2.png
    Ignored:    analysis/Study_3.png
    Ignored:    analysis/Study_4A.png
    Ignored:    analysis/Study_4B.png
    Ignored:    code/.RData
    Ignored:    code/.Rhistory

Untracked files:
    Untracked:  analysis/004_throughput_analysis.Rmd
    Untracked:  analysis/005_mre_initialization.Rmd
    Untracked:  analysis/007_phylosmith_abundance.Rmd
    Untracked:  analysis/008_phylosmith_diversity.Rmd
    Untracked:  analysis/009_quality_control.Rmd
    Untracked:  analysis/100_global_description.Rmd
    Untracked:  analysis/101_baseline_timepoints.Rmd
    Untracked:  analysis/102_starting_point_description.Rmd
    Untracked:  analysis/103_baseline_description.Rmd
    Untracked:  analysis/104_groups_comparison.Rmd
    Untracked:  analysis/105_longitudinal_sudy_t1.Rmd
    Untracked:  analysis/105_longitudinal_sudy_t2.Rmd
    Untracked:  analysis/105_longitudinal_sudy_t3.Rmd
    Untracked:  analysis/105_longitudinal_sudy_t4A.Rmd
    Untracked:  analysis/105_longitudinal_sudy_t4B.Rmd
    Untracked:  analysis/Study.jpg
    Untracked:  code/000_s3_bucket_initialization.Rmd
    Untracked:  code/001_metadata_check.Rmd
    Untracked:  code/002_datatransfer.Rmd
    Untracked:  code/003_metadata_vars_selection.Rmd
    Untracked:  code/006_populate_mre.Rmd
    Untracked:  data/RV306_CVM_2022_11_08_lims.xlsx
    Untracked:  data/RV306_CVM_ShippingManifest_2022_SB_clinical.xlsx
    Untracked:  data/preprocessed/

Unstaged changes:
    Modified:   analysis/_site.yml
    Modified:   analysis/about.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="vars-path-definitions-libs" class="section level2">
<h2>Vars &amp; path definitions &amp; libs</h2>
<pre class="r"><code>library(magrittr)

bucket &lt;- &quot;s3://cihr-hiv-rv306&quot;
filt_data_obj &lt;- &quot;vaginal/metagenome/WMGS/RawData/DataFiltering_Boxplot.rds&quot;</code></pre>
</div>
<div id="throughput-analysis" class="section level2">
<h2>Throughput analysis</h2>
<p>This is a pre-analysis of <strong>CHIR HIV RV306 Vaginal</strong>
Shotgun data, aimed to assess the bacterial yield in the raw data in
order to be able to make informed decisions downstream and on other
projects with similar samples.</p>
<div id="load-and-preprocessing-data" class="section level4">
<h4>Load and Preprocessing data</h4>
<pre class="r"><code>## Load data
df &lt;- 
  aws.s3::s3readRDS(filt_data_obj, bucket) %&gt;%
  .[[1]] %&gt;%
  tibble::as_tibble()

df</code></pre>
<pre><code>#&gt; # A tibble: 1,443 × 3
#&gt;    SampleID Step          value
#&gt;    &lt;ord&gt;    &lt;ord&gt;         &lt;int&gt;
#&gt;  1 S014117  QFailReads  1002747
#&gt;  2 S014255  QFailReads 13565260
#&gt;  3 S014196  QFailReads   934245
#&gt;  4 S014207  QFailReads   984436
#&gt;  5 S014287  QFailReads  1311930
#&gt;  6 S014521  QFailReads   885726
#&gt;  7 S014264  QFailReads  3297456
#&gt;  8 S014223  QFailReads  1164790
#&gt;  9 S014288  QFailReads  2204796
#&gt; 10 S014128  QFailReads   724235
#&gt; # ℹ 1,433 more rows</code></pre>
<pre class="r"><code>## Preprocessing of data
df &lt;- 
  df %&gt;% 
  tidyr::pivot_wider(id_cols = SampleID, names_from = Step, values_from = value) %&gt;%
  dplyr::rowwise() %&gt;% 
  dplyr::mutate(TotalReads = sum(dplyr::c_across(where(is.numeric)))) %&gt;%
  tidyr::pivot_longer(names_to = &quot;Step&quot;, cols = where(is.numeric)) %&gt;%
  dplyr::arrange(SampleID) %&gt;%
  dplyr::mutate(
    Step = forcats::fct_relevel(Step, c(&quot;TotalReads&quot;, &quot;Human&quot;, &quot;QFailReads&quot;, &quot;QPassReads&quot;)), 
    counts = value
  )</code></pre>
</div>
<div id="absolute-counts" class="section level4">
<h4>Absolute Counts (%)</h4>
<pre class="r"><code>abs_plt &lt;- ggstatsplot::ggwithinstats(
  df,
  x = Step,
  y = counts,
  pairwise.display = &quot;none&quot;,
  title = &quot;Absolute Throughput&quot;,
  subtitle = NULL 
)</code></pre>
<pre><code>#&gt; Error in if (...length() &amp;&amp; any(...names() == &quot;Dimnames&quot;)) .Object@Dimnames &lt;- fixupDN(.Object@Dimnames) : 
#&gt;   missing value where TRUE/FALSE needed</code></pre>
<pre class="r"><code>abs_means &lt;- aggregate(counts ~ Step, data = df, FUN = mean)
abs_medians &lt;- aggregate(counts ~ Step, data = df, FUN = median)

abs_plt</code></pre>
<p><img src="figure/004_throughput_analysis.Rmd/absolute_counts_plot-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="relative-counts" class="section level4">
<h4>Relative Counts</h4>
<pre class="r"><code>df1 &lt;- df %&gt;%
  tidyr::pivot_wider(id_cols = SampleID, names_from = Step, values_from = value) %&gt;% 
  dplyr::mutate(dplyr::across(where(is.numeric), ~ .x / TotalReads * 100)) %&gt;%
  tidyr::pivot_longer(names_to = &quot;Step&quot;, cols = where(is.numeric)) %&gt;%
  dplyr::mutate(
    Step = forcats::fct_relevel(Step, c(&quot;TotalReads&quot;, &quot;Human&quot;, &quot;QFailReads&quot;, &quot;QPassReads&quot;)),
    counts = value
  )

rel_plt &lt;-df1%&gt;% 
  ggstatsplot::ggwithinstats(
    x = Step,
    y = counts,
    pairwise.display = &quot;none&quot;,
    title = &quot;Relative Throughput&quot;,
    subtitle = NULL
  )</code></pre>
<pre><code>#&gt; Error in if (...length() &amp;&amp; any(...names() == &quot;Dimnames&quot;)) .Object@Dimnames &lt;- fixupDN(.Object@Dimnames) : 
#&gt;   missing value where TRUE/FALSE needed</code></pre>
<pre class="r"><code>rel_means &lt;- aggregate(counts ~ Step, data = df1, FUN = mean)
rel_medians &lt;- aggregate(counts ~ Step, data = df1, FUN = median)

rel_plt</code></pre>
<p><img src="figure/004_throughput_analysis.Rmd/relative_counts_plot-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="numerical-analysis" class="section level4">
<h4>Numerical Analysis</h4>
<pre class="r"><code>df %&gt;% 
  dplyr::filter(Step == &quot;QPassReads&quot;) %&gt;%
  dplyr::pull(value) %&gt;%
  stats::quantile()</code></pre>
<pre><code>#&gt;       0%      25%      50%      75%     100% 
#&gt;   117462   973315  2323862  5110044 25110552</code></pre>
</div>
<div id="interpretation" class="section level4">
<h4>Interpretation</h4>
<p>As expected, there is a large percentage of reads coming from human
content (mean value = ~77%, similar to Flamingo dataset). Quality pass
reads are low with a mean value of approximately 4M reads but a median
value of around 2M reads. This will probably affect our taxonomical
characterization capabilities using marker-gene approaches, since by
definition they only use a small portion of the QPass reads. Thus,
non-dominant species may be frequently missed and general taxonomical
characterization may be poor.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>#&gt; R version 4.1.2 (2021-11-01)
#&gt; Platform: x86_64-pc-linux-gnu (64-bit)
#&gt; Running under: Ubuntu 22.04.4 LTS
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so
#&gt; 
#&gt; locale:
#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#&gt;  [3] LC_TIME=es_ES.UTF-8        LC_COLLATE=en_US.UTF-8    
#&gt;  [5] LC_MONETARY=es_ES.UTF-8    LC_MESSAGES=en_US.UTF-8   
#&gt;  [7] LC_PAPER=es_ES.UTF-8       LC_NAME=C                 
#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
#&gt; [11] LC_MEASUREMENT=es_ES.UTF-8 LC_IDENTIFICATION=C       
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     
#&gt; 
#&gt; other attached packages:
#&gt; [1] magrittr_2.0.3  workflowr_1.7.1
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;   [1] nlme_3.1-164           fs_1.6.3               insight_0.19.8        
#&gt;   [4] httr_1.4.7             rprojroot_2.0.4        numDeriv_2016.8-1.1   
#&gt;   [7] tools_4.1.2            bslib_0.7.0            utf8_1.2.4            
#&gt;  [10] R6_2.5.1               afex_1.3-1             statsExpressions_1.5.3
#&gt;  [13] colorspace_2.1-0       withr_3.0.0            tidyselect_1.2.1      
#&gt;  [16] processx_3.8.4         curl_5.2.1             compiler_4.1.2        
#&gt;  [19] git2r_0.33.0           cli_3.6.2              xml2_1.3.6            
#&gt;  [22] prismatic_1.1.1        labeling_0.4.3         bayestestR_0.13.2     
#&gt;  [25] sass_0.4.9             scales_1.3.0           mvtnorm_1.2-4         
#&gt;  [28] callr_3.7.6            pbapply_1.7-2          stringr_1.5.1         
#&gt;  [31] digest_0.6.35          minqa_1.2.6            rmarkdown_2.26        
#&gt;  [34] base64enc_0.1-3        pkgconfig_2.0.3        htmltools_0.5.8.1     
#&gt;  [37] lme4_1.1-35.1          highr_0.10             fastmap_1.1.1         
#&gt;  [40] rlang_1.1.3            rstudioapi_0.15.0      farver_2.1.1          
#&gt;  [43] jquerylib_0.1.4        generics_0.1.3         jsonlite_1.8.8        
#&gt;  [46] dplyr_1.1.4            car_3.1-2              parameters_0.21.5     
#&gt;  [49] patchwork_1.2.0        Matrix_1.6-5           Rcpp_1.0.12           
#&gt;  [52] munsell_0.5.1          fansi_1.0.6            abind_1.4-5           
#&gt;  [55] lifecycle_1.0.4        stringi_1.8.3          whisker_0.4.1         
#&gt;  [58] yaml_2.3.8             carData_3.0-5          MASS_7.3-60.0.1       
#&gt;  [61] plyr_1.8.9             grid_4.1.2             paletteer_1.6.0       
#&gt;  [64] parallel_4.1.2         promises_1.3.0         ggrepel_0.9.5         
#&gt;  [67] forcats_1.0.0          lattice_0.22-5         splines_4.1.2         
#&gt;  [70] zeallot_0.1.0          knitr_1.46             ps_1.7.6              
#&gt;  [73] pillar_1.9.0           boot_1.3-30            effectsize_0.8.6      
#&gt;  [76] reshape2_1.4.4         glue_1.7.0             evaluate_0.23         
#&gt;  [79] getPass_0.2-4          vctrs_0.6.5            nloptr_2.0.3          
#&gt;  [82] httpuv_1.6.14          MatrixModels_0.5-3     gtable_0.3.4          
#&gt;  [85] aws.s3_0.3.21          purrr_1.0.2            BayesFactor_0.9.12-4.7
#&gt;  [88] tidyr_1.3.1            rematch2_2.1.2         datawizard_0.9.1      
#&gt;  [91] cachem_1.0.8           ggplot2_3.5.0          xfun_0.43             
#&gt;  [94] correlation_0.8.4      coda_0.19-4.1          later_1.3.2           
#&gt;  [97] tibble_3.2.1           lmerTest_3.1-3         ggstatsplot_0.12.2    
#&gt; [100] aws.signature_0.6.0    here_1.0.1</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
