<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Francesc Català-Moll &amp; Oriol Careta Borràs" />

<meta name="date" content="2024-06-04" />

<title>CHIR HIV RV306 Vaginal - mre Initialization</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">2024_cihr_hiv_rv306</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-check-square"></span>
     
    Quality control and initial microbiome exploration
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="004_throughput_analysis.html">Throughput Analysis</a>
    </li>
    <li>
      <a href="005_mre_initialization.html">Init mre object</a>
    </li>
    <li>
      <a href="007_phylosmith_abundance.html">Phylosmith abundance exploration</a>
    </li>
    <li>
      <a href="008_phylosmith_diversity.html">Phylosmith diversity exploration</a>
    </li>
    <li>
      <a href="009_quality_control.html">Quality_control</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-flag-checkered"></span>
     
    Analysis of baseline
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="100_global_description.html">Global description</a>
    </li>
    <li>
      <a href="101_baseline_timepoints.html">Baseline timepoints</a>
    </li>
    <li>
      <a href="102_starting_point_description.html">Starting point description</a>
    </li>
    <li>
      <a href="103_baseline_description.html">Baseline description</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-flask"></span>
     
    Analysis of vaccination effects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="104_groups_comparison.html">Groups Comparison</a>
    </li>
    <li>
      <a href="105_longitudinal_sudy_t1.html">Longitudinal Study of Group 1</a>
    </li>
    <li>
      <a href="105_longitudinal_sudy_t2.html">Longitudinal Study of Group 2</a>
    </li>
    <li>
      <a href="105_longitudinal_sudy_t3.html">Longitudinal Study of Group 3</a>
    </li>
    <li>
      <a href="105_longitudinal_sudy_t4A.html">Longitudinal Study of Group 4A</a>
    </li>
    <li>
      <a href="105_longitudinal_sudy_t4B.html">Longitudinal Study of Group 4B</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">CHIR HIV RV306 Vaginal - mre
Initialization</h1>
<h4 class="author">Francesc Català-Moll &amp; Oriol Careta Borràs</h4>
<h4 class="date">2024-06-04</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-06-04
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>OriolGEM.github.io/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240321code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240321)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240321code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240321)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomOriolGEMOriolGEMgithubiotree64c7378a0d3b7cd42b953f017d91ed4ba18aca01targetblank64c7378a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/tree/64c7378a0d3b7cd42b953f017d91ed4ba18aca01" target="_blank">64c7378</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomOriolGEMOriolGEMgithubiotree64c7378a0d3b7cd42b953f017d91ed4ba18aca01targetblank64c7378a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/OriolGEM/OriolGEM.github.io/tree/64c7378a0d3b7cd42b953f017d91ed4ba18aca01" target="_blank">64c7378</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/.RData
    Ignored:    code/.Rhistory
    Ignored:    data/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/005_mre_initialization.Rmd</code>) and HTML
(<code>docs/005_mre_initialization.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/analysis/005_mre_initialization.Rmd" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/OriolGEM/OriolGEM.github.io/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/005_mre_initialization.html" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>This code obtains different data from different sources:</p>
<ul>
<li>Taxonomical Composition data derived from
Metaphlan3/motus2/kraken2/dada2 analysis. This data is obtained through
the execution of the AWS/Batch/WMGS pipeline as standard R::phyloseq
objects</li>
<li>Gene Function Data, derived from FMAP/HUMANN3/IGC/VIRGO catalogs or
tools. This data is obtained through the execution of the AWS/Batch/WMGS
pipeline as standard R::phyloseq objects</li>
<li>Gene Richness or diversity, obtained from either IGC/VIRGO catalogs
or dada2.</li>
<li>Metadata, obtained from the study data source. Metadata is probably
formatted and curated beforehand and linked to taxonomic and function
data during the construction of the two previous data sources.</li>
<li>Other: In this code, it is the moment to alter the original data
sources in any way: modification, removal, or addition of
variables.</li>
</ul>
<p>Data is obtained, merged and provided a consistent starting point for
statistical analysis. Metadata is also described separately to produce a
metadata-only report.</p>
<p>Finally, a MRE object is created that is used downstream as the
starting point for all analysis and (automatic) reporting.</p>
</div>
<div id="vars-path-definitions-libs" class="section level2">
<h2>Vars &amp; path definitions &amp; libs</h2>
<pre class="r"><code>library(magrittr)

bucket &lt;- &quot;s3://cihr-hiv-rv306/vaginal/metagenome/WMGS&quot;
out_dir &lt;- here::here(&quot;data/preprocessed/&quot;)
dir.create(out_dir, showWarnings = FALSE, recursive = TRUE)</code></pre>
</div>
<div id="mre-object-initialization" class="section level2">
<h2>MRE object initialization</h2>
<p>A metar (MRE) object contains different slots, each one containing
different types of data for the study being analysed. A brief
description of the available data for this project is:</p>
<pre class="r"><code>mre &lt;- metar::metarExperiment_aws(bucket = bucket, wd = out_dir)

mre</code></pre>
<pre><code>#&gt; MetarSet-class experimental-level object 
#&gt;   [MetadataSet     ]: 481 samples and 6 variables 
#&gt;   [DiversitySet    ]: 
#&gt;      ✖ igc: 0 observations and 0 samples 
#&gt;      ✔ virgo: 14765 observations and 481 samples 
#&gt;      ✖ dada2: 0 observations and 0 samples 
#&gt;   [TaxaSet         ]: 
#&gt;      ✖ metaphlan: no phyloseq object available 
#&gt;      ✖ kraken: no phyloseq object available 
#&gt;      ✖ bracken: no phyloseq object available 
#&gt;      ✖ motus: no phyloseq object available 
#&gt;      ✔ virgo: phyloseq object with 269 taxa and 481 samples 
#&gt;      ✖ dada2: no phyloseq object available 
#&gt;   [GeneFunctionSet ]: 
#&gt;      ✖ humann: no functional data aviable 
#&gt;      ✖ igc: no functional data aviable 
#&gt;      ✔ virgo: 4621 kegg_id, 0 cog_id, 0 eggnog_id, 0 ec_id and 0 metacyc_id 
#&gt;      ✖ fmap: no functional data aviable 
#&gt;      ✖ picrust2: no functional data aviable 
#&gt;      ✖ tax4fun: no functional data aviable 
#&gt;      ✖ phylolisft: no functional data aviable</code></pre>
<p>It can be seen that there are only data obtained from the VIRGO
catalog, as it is the best tool to analyse human vaginal microbiome.</p>
</div>
<div id="mre-metadata-modificatiopn" class="section level2">
<h2>MRE metadata modificatiopn</h2>
<pre class="r"><code># Create new dataframe with the variable to add
df &lt;- mre@metadata@metadata_df
new_df &lt;- df %&gt;% 
  dplyr::select(SampleID, Treatment_group) %&gt;% 
  dplyr::mutate(
    Treatment_group_1 = dplyr::case_when(
      Treatment_group %in% &quot;t_4&quot; ~ &quot;t_4A&quot;,
      Treatment_group %in% &quot;t_no_boost&quot; ~ &quot;t_4A&quot;,
      TRUE ~ as.character (Treatment_group)
    )
  ) %&gt;% 
  dplyr::select(SampleID, Treatment_group_1)

# Add the new var to the mre
mre &lt;- metar::add_metadata_var(mre, new_df, id_col = &quot;SampleID&quot;)

# Add the new var as categorical var
mre &lt;- metar::add_cat(mre, &quot;Treatment_group_1&quot;, &quot;Set1&quot;)</code></pre>
</div>
<div id="save-mre" class="section level2">
<h2>Save MRE</h2>
<p>The MRE object will be saved with the name “mre.rds” if no previous
file exists, but as the date plus mre.rds (as an example,
“20241231_mre.rds”) if it already exists (as the file called “mre.rds”
will already be populated by a set of multiple analysis). Furthermore,
any previous file with a name consistent of a previous date plus mre.rds
(following the example, “20241230_mre.rds”) will be removed.</p>
<pre class="r"><code>existing_mre_files &lt;- list.files(path = out_dir, pattern = &quot;^\\d{8}_mre\\.rds$&quot;)

if (length(existing_mre_files) &gt; 0) {
  file.remove(file.path(out_dir, existing_mre_files))
}</code></pre>
<pre><code>#&gt; [1] TRUE</code></pre>
<pre class="r"><code>if (&quot;mre.rds&quot; %in% list.files(path = out_dir)) {
  today &lt;- format(Sys.Date(), &quot;%Y%m%d&quot;)
  filename &lt;- paste0(today, &quot;_mre.rds&quot;)
} else {
  filename &lt;- &quot;mre.rds&quot;
}

readr::write_rds(mre, file = here::here(out_dir, filename))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>#&gt; R version 4.1.2 (2021-11-01)
#&gt; Platform: x86_64-pc-linux-gnu (64-bit)
#&gt; Running under: Ubuntu 22.04.4 LTS
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so
#&gt; 
#&gt; locale:
#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#&gt;  [3] LC_TIME=es_ES.UTF-8        LC_COLLATE=en_US.UTF-8    
#&gt;  [5] LC_MONETARY=es_ES.UTF-8    LC_MESSAGES=en_US.UTF-8   
#&gt;  [7] LC_PAPER=es_ES.UTF-8       LC_NAME=C                 
#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
#&gt; [11] LC_MEASUREMENT=es_ES.UTF-8 LC_IDENTIFICATION=C       
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     
#&gt; 
#&gt; other attached packages:
#&gt; [1] magrittr_2.0.3
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;   [1] nlme_3.1-164           bitops_1.0-7           fs_1.6.4              
#&gt;   [4] phyloseq_1.38.0        bit64_4.0.5            httr_1.4.7            
#&gt;   [7] rprojroot_2.0.4        GenomeInfoDb_1.30.1    tools_4.1.2           
#&gt;  [10] bslib_0.7.0            utf8_1.2.4             R6_2.5.1              
#&gt;  [13] vegan_2.6-6.1          lazyeval_0.2.2         BiocGenerics_0.40.0   
#&gt;  [16] mgcv_1.9-1             colorspace_2.1-0       permute_0.9-7         
#&gt;  [19] rhdf5filters_1.6.0     ade4_1.7-22            withr_3.0.0           
#&gt;  [22] tidyselect_1.2.1       bit_4.0.5              curl_5.2.1            
#&gt;  [25] compiler_4.1.2         git2r_0.33.0           cli_3.6.2             
#&gt;  [28] Biobase_2.54.0         xml2_1.3.6             plotly_4.10.4         
#&gt;  [31] sass_0.4.9             scales_1.3.0           readr_2.1.5           
#&gt;  [34] stringr_1.5.1          digest_0.6.35          rmarkdown_2.27        
#&gt;  [37] XVector_0.34.0         base64enc_0.1-3        pkgconfig_2.0.3       
#&gt;  [40] htmltools_0.5.8.1      fastmap_1.2.0          htmlwidgets_1.6.4     
#&gt;  [43] rlang_1.1.3            rstudioapi_0.16.0      jquerylib_0.1.4       
#&gt;  [46] generics_0.1.3         jsonlite_1.8.8         vroom_1.6.5           
#&gt;  [49] dplyr_1.1.4            RCurl_1.98-1.14        GenomeInfoDbData_1.2.7
#&gt;  [52] biomformat_1.22.0      Matrix_1.3-4           Rcpp_1.0.12           
#&gt;  [55] munsell_0.5.1          S4Vectors_0.32.4       Rhdf5lib_1.16.0       
#&gt;  [58] fansi_1.0.6            logger_0.3.0           ape_5.8               
#&gt;  [61] lifecycle_1.0.4        stringi_1.8.4          whisker_0.4.1         
#&gt;  [64] yaml_2.3.8             MASS_7.3-60.0.1        zlibbioc_1.40.0       
#&gt;  [67] rhdf5_2.38.1           plyr_1.8.9             grid_4.1.2            
#&gt;  [70] parallel_4.1.2         promises_1.3.0         ggrepel_0.9.5         
#&gt;  [73] crayon_1.5.2           lattice_0.22-6         Biostrings_2.62.0     
#&gt;  [76] splines_4.1.2          multtest_2.50.0        hms_1.1.3             
#&gt;  [79] knitr_1.46             pillar_1.9.0           igraph_2.0.3          
#&gt;  [82] reshape2_1.4.4         codetools_0.2-20       stats4_4.1.2          
#&gt;  [85] glue_1.7.0             evaluate_0.23          metar_0.1.5           
#&gt;  [88] data.table_1.15.4      tzdb_0.4.0             vctrs_0.6.5           
#&gt;  [91] httpuv_1.6.15          foreach_1.5.2          gtable_0.3.5          
#&gt;  [94] aws.s3_0.3.21          purrr_1.0.2            tidyr_1.3.1           
#&gt;  [97] cachem_1.1.0           ggplot2_3.5.1          xfun_0.44             
#&gt; [100] later_1.3.2            survival_3.6-4         viridisLite_0.4.2     
#&gt; [103] ggmosaic_0.3.3         tibble_3.2.1           iterators_1.0.14      
#&gt; [106] aws.signature_0.6.0    IRanges_2.28.0         workflowr_1.7.1       
#&gt; [109] cluster_2.1.6          here_1.0.1</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
