<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Francesc Català-Moll &amp; Oriol Careta Borràs" />

<meta name="date" content="2024-06-04" />

<title>CHIR HIV RV306 Vaginal - Quality Control</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">2024_cihr_hiv_rv306</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-check-square"></span>
     
    Quality control and initial microbiome exploration
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="004_throughput_analysis.html">Throughput Analysis</a>
    </li>
    <li>
      <a href="005_mre_initialization.html">Init mre object</a>
    </li>
    <li>
      <a href="007_phylosmith_abundance.html">Phylosmith abundance exploration</a>
    </li>
    <li>
      <a href="008_phylosmith_diversity.html">Phylosmith diversity exploration</a>
    </li>
    <li>
      <a href="009_quality_control.html">Quality_control</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-flag-checkered"></span>
     
    Analysis of baseline
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="100_global_description.html">Global description</a>
    </li>
    <li>
      <a href="101_baseline_timepoints.html">Baseline timepoints</a>
    </li>
    <li>
      <a href="102_starting_point_description.html">Starting point description</a>
    </li>
    <li>
      <a href="103_baseline_description.html">Baseline description</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-flask"></span>
     
    Analysis of vaccination effects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="104_groups_comparison.html">Groups Comparison</a>
    </li>
    <li>
      <a href="105_longitudinal_sudy_t1.html">Longitudinal Study of Group 1</a>
    </li>
    <li>
      <a href="105_longitudinal_sudy_t2.html">Longitudinal Study of Group 2</a>
    </li>
    <li>
      <a href="105_longitudinal_sudy_t3.html">Longitudinal Study of Group 3</a>
    </li>
    <li>
      <a href="105_longitudinal_sudy_t4A.html">Longitudinal Study of Group 4A</a>
    </li>
    <li>
      <a href="105_longitudinal_sudy_t4B.html">Longitudinal Study of Group 4B</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">CHIR HIV RV306 Vaginal - Quality
Control</h1>
<h4 class="author">Francesc Català-Moll &amp; Oriol Careta Borràs</h4>
<h4 class="date">2024-06-04</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-06-04
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>OriolGEM.github.io/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240321code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240321)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240321code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240321)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomOriolGEMOriolGEMgithubiotree64c7378a0d3b7cd42b953f017d91ed4ba18aca01targetblank64c7378a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/tree/64c7378a0d3b7cd42b953f017d91ed4ba18aca01" target="_blank">64c7378</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomOriolGEMOriolGEMgithubiotree64c7378a0d3b7cd42b953f017d91ed4ba18aca01targetblank64c7378a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/OriolGEM/OriolGEM.github.io/tree/64c7378a0d3b7cd42b953f017d91ed4ba18aca01" target="_blank">64c7378</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/.RData
    Ignored:    code/.Rhistory
    Ignored:    data/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/009_quality_control.Rmd</code>)
and HTML (<code>docs/009_quality_control.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/analysis/009_quality_control.Rmd" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/OriolGEM/OriolGEM.github.io/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/009_quality_control.html" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>We are interested in a general description of the dataset in terms
quality control and also of microbiome composition and function. We are
using all available samples (481) from all patients (118).</p>
</div>
<div id="load-mre" class="section level2">
<h2>Load mre</h2>
<pre class="r"><code>## mre init &amp; filter
mre &lt;- readr::read_rds(here::here(&quot;data&quot;, &quot;preprocessed&quot;, &quot;mre.rds&quot;))</code></pre>
</div>
<div id="qc" class="section level2">
<h2>QC</h2>
<pre class="r"><code>## def vars
bucket &lt;- &quot;s3://cihr-hiv-rv306&quot;
filt_data_obj &lt;- &quot;vaginal/metagenome/WMGS/RawData/DataFiltering_Boxplot.rds&quot;

## load seq analysis data
df &lt;- 
  aws.s3::s3readRDS(filt_data_obj, bucket) %&gt;%
  .[[1]] %&gt;%
  tibble::as_tibble()

# generate metadataDF from mre
metadataDF &lt;- data.frame(mre@metadata@metadata_df)

# plot longitudinal sampling per each Arm
metadataDF$Collection_date &lt;- as.Date(metadataDF$Collection_date)
ggplot(metadataDF, aes(x = Collection_date, y = PatientID, color = Treatment_group_1, shape = Arm)) +
  geom_point(size = 1.5) +
  geom_line() +
  labs(x = &quot;Date&quot;, y = &quot;ID&quot;) +
  scale_shape_manual(name = &#39;Arm&#39;, values = c(&#39;Placebo&#39; = 1, &#39;Active&#39; = 2)) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/009_quality_control.Rmd/treatment_group_followup-1.png" width="2700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-treatment_group_followup-1">
Past versions of treatment_group_followup-1.png
</button>
</p>
<div id="fig-treatment_group_followup-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/009_quality_control.Rmd/treatment_group_followup-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The dates of the collection of each sample can be seen here. It can
be seen that depending on the treatment group, samples were collected at
different time-points.</p>
<pre class="r"><code>con1 &lt;- table(metadataDF$Medical_center, metadataDF$Visit_Week)
addmargins(con1)</code></pre>
<pre><code>#&gt;      
#&gt;       Wk 0 Wk 14 Wk 26 Wk 50 Wk 62 Wk 74 Sum
#&gt;   AAA  109   104   100    99    18    14 444
#&gt;   MUW    9     9     9     9     1     0  37
#&gt;   Sum  118   113   109   108    19    14 481</code></pre>
<pre class="r"><code># Create a ggplot object
p &lt;- ggplot(metadataDF, aes(x = Visit_Week, fill = Medical_center)) +
  geom_bar(position = &quot;dodge&quot;) +
  facet_wrap(~ Arm) +
  labs(title = &quot;Histogram of Medical Center Visits by Week&quot;,
       x = &quot;Visit Week&quot;,
       y = &quot;Number of patients&quot;) +
  scale_fill_manual(values = c(&quot;deeppink4&quot;, &quot;darkslategray4&quot;)) +
  theme_minimal() +
  theme(panel.grid.major = element_blank(),
      panel.grid.minor = element_blank())

p</code></pre>
<p><img src="figure/009_quality_control.Rmd/medical_center_all-1.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-medical_center_all-1">
Past versions of medical_center_all-1.png
</button>
</p>
<div id="fig-medical_center_all-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/009_quality_control.Rmd/medical_center_all-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We can also see the number of patient’s samples extracted each week,
separated by medical center. There is a drop in number of patient’s
samples extracted between week 50 and week 62, because groups 1, 2 and 3
had samples taken only up until week 50. We can also observe that there
are very few patients on the placebo arm.</p>
<pre class="r"><code># filter seq analysis data
sub_df &lt;- dplyr::filter(df, SampleID %in% metadataDF$SampleID)

# dcast sub_df and merge with metadata
sub_df_dcast &lt;- reshape2::dcast(sub_df, SampleID ~ Step, value.var = &quot;value&quot;)
all_df &lt;- base::merge(metadataDF, sub_df_dcast, by = &quot;SampleID&quot;)

all_df$Visit_Week &lt;- gsub(&#39;Wk &#39;, &#39;wk&#39;, all_df$Visit_Week)

all_df$Concentration_ng_ul[all_df$Concentration_ng_ul == 10000.0] &lt;- 100
all_df2 &lt;-
  all_df %&gt;% dplyr::select(
    SampleID,
    PatientID,
    Visit_Week,
    Arm,
    QFailReads,
    Human,
    QPassReads
  )

# melt df_all
all_df_melt &lt;- reshape2::melt(all_df2)

# barplot
ggplot(all_df_melt, aes(x = SampleID, y = value, fill = variable)) +
  geom_bar(stat = &quot;identity&quot;) +
  ylab(&quot;PE reads&quot;) +
  theme_bw() +
  scale_fill_manual(values = c(&quot;azure3&quot;, &quot;darkslateblue&quot;, &quot;yellow&quot;)) +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 3),
    axis.title.x = element_blank()
  )</code></pre>
<p><img src="figure/009_quality_control.Rmd/plot_abs_qreads-1.png" width="2700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot_abs_qreads-1">
Past versions of plot_abs_qreads-1.png
</button>
</p>
<div id="fig-plot_abs_qreads-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/009_quality_control.Rmd/plot_abs_qreads-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here we can see the total number of pair-ended reads (PE reads) of
each sample separated by QFail reads, QPass reads and Human reads.</p>
<ul>
<li><p>QFail reads: These are sequencing reads that have failed quality
control (QC) metrics or filters during the sequencing process. This
could be due to low sequencing quality, adapter contamination, or other
technical issues.</p></li>
<li><p>QPass reads: These are sequencing reads that have passed quality
control metrics and are considered to meet the quality standards set for
the sequencing run.</p></li>
<li><p>Human reads: These are sequencing reads that have passed quality
control metrics and align to the human genome.</p></li>
</ul>
<pre class="r"><code># Summary QPass N reads
x &lt;- summary(all_df$QPassReads)
x</code></pre>
<pre><code>#&gt;     Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
#&gt;   117462   973315  2323862  4018564  5110044 25110552</code></pre>
<p>The summary statics of the QPass reads can be seen here. Quality pass
reads are low with a mean value of approximately 4M reads but a median
value of around 2M.</p>
<pre class="r"><code># plot qPass reads for each arm
ggplot(all_df, aes(x = reorder(PatientID, +QPassReads), y = QPassReads)) +
  geom_bar(aes(fill = Arm), stat = &quot;identity&quot;) +
  ylab(&quot;QPass reads&quot;) +
  facet_wrap( ~ Visit_Week) +
  scale_fill_manual(values = c(&quot;azure3&quot;, &quot;darkslateblue&quot;)) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 3),
    axis.title.x = element_blank()
  )</code></pre>
<p><img src="figure/009_quality_control.Rmd/plot_Qpassreads_arm-1.png" width="2700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot_Qpassreads_arm-1">
Past versions of plot_Qpassreads_arm-1.png
</button>
</p>
<div id="fig-plot_Qpassreads_arm-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/009_quality_control.Rmd/plot_Qpassreads_arm-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here we can see the total number of QPass reads depending if they
belong to the active or placebo arm, separated by time-point. The number
of QPass reads is very variable among samples but it does not seem that
there is any correlation between the number of QPass reads and
time-point or arm.</p>
<pre class="r"><code>ggplot(all_df, aes(x = Arm, y = QPassReads, color = Arm)) +
  geom_boxplot(size = 0.5, outlier.size = 0.9, alpha = 0.8) +
  geom_point(size = 0.3, color = &quot;gray35&quot;) +
  ylab(&quot;QPass reads&quot;) +
  scale_fill_manual(values = c(&quot;snow&quot;)) +
  scale_color_manual(values = c(&quot;deeppink4&quot;, &quot;darkslategray4&quot;)) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 8),
    axis.title.x = element_blank()
  ) + 
  ggpubr::stat_compare_means()</code></pre>
<p><img src="figure/009_quality_control.Rmd/boxplot_Qpassreads_arm-1.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-boxplot_Qpassreads_arm-1">
Past versions of boxplot_Qpassreads_arm-1.png
</button>
</p>
<div id="fig-boxplot_Qpassreads_arm-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/009_quality_control.Rmd/boxplot_Qpassreads_arm-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here we can see the total number of QPass reads depending if they
belong to the active or placebo group, there are no differences between
groups.</p>
<pre class="r"><code># correlation DNA concentration and final N reads
ggpubr::ggscatter(
  all_df,
  y = &quot;Concentration_ng_ul&quot;,
  x = &quot;QPassReads&quot;,
  color = &quot;blue&quot;,
  size = 2,
  add = &quot;reg.line&quot;,
  conf.int = TRUE,
  cor.coef = FALSE,
  cor.method = &quot;spearman&quot;,
  add.params = list(color = &quot;gray25&quot;, fill = &quot;gray75&quot;),
  # Customize reg. line
  ylab = &quot;DNA&quot;,
  xlab = &quot;QPass reads&quot;
) +
  ggpubr::stat_cor(method = &quot;spearman&quot;, label.x = 0, label.y = 150)</code></pre>
<p><img src="figure/009_quality_control.Rmd/DNA_qPass-1.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-DNA_qPass-1">
Past versions of DNA_qPass-1.png
</button>
</p>
<div id="fig-DNA_qPass-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/009_quality_control.Rmd/DNA_qPass-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here we can see a scatter plot on the DNA concentration against the
number of QPass reads. The spearmann correlation indicates that there is
some negative correlation between the DNA concentration and the number
of reads that have passed the QC control. At higher DNA concentration,
less number of QPass reads. This is because a higher DNA concentration
is most of the times due to the presence of contamination.</p>
<pre class="r"><code>## Plot absolute
df %&gt;%
  ggplot(aes(SampleID, value, fill = Step)) +
  geom_bar(position = &quot;stack&quot;, stat = &quot;identity&quot;, alpha = 0.8, colour = 0.0, width = 0.8) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  theme_minimal() +
  theme(axis.text.x = element_blank())</code></pre>
<p><img src="figure/009_quality_control.Rmd/absolute_counts-1.png" width="2700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-absolute_counts-1">
Past versions of absolute_counts-1.png
</button>
</p>
<div id="fig-absolute_counts-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/009_quality_control.Rmd/absolute_counts-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here we can see the total number of pair-ended reads (PE reads) of
each sample separated by QFail reads, QPass reads and Human reads.
Samples appear ordered depending on the amount of Qpass reads. This plot
is useful to determine a threshold of QPass reads to determine which
samples to use on posterior analyses. The proposal now is a minimum
number of 100,000 QPass reads per sample.</p>
<p>It has been suggested by bibliography that it is best to perform a
cutoff of the samples where the max number of mapped reads are less than
100,000. I we do so, we lose 7 samples:</p>
<pre class="r"><code>mre@diversity@virgo@dataTable %&gt;%
  dplyr::group_by(SampleID) %&gt;%
  dplyr::summarise(NumberMappedReads = max(NumberMappedReads)) %&gt;%
  dplyr::filter(NumberMappedReads &lt; 100000)</code></pre>
<pre><code>#&gt; # A tibble: 7 × 2
#&gt;   SampleID NumberMappedReads
#&gt;   &lt;chr&gt;                &lt;int&gt;
#&gt; 1 S014117              54803
#&gt; 2 S014196              58423
#&gt; 3 S014255              63120
#&gt; 4 S014264              84807
#&gt; 5 S014287              60104
#&gt; 6 S014288              78139
#&gt; 7 S014521              53695</code></pre>
<pre class="r"><code>selected_samples &lt;- c(&quot;S014117&quot;, &quot;S014196&quot;, &quot;S014255&quot;, &quot;S014264&quot;, &quot;S014287&quot;, &quot;S014288&quot;, &quot;S014521&quot;)

df %&gt;%
  ggplot(aes(SampleID, value, fill = Step)) +
  geom_bar(position = &quot;stack&quot;, stat = &quot;identity&quot;, alpha = 0.8, colour = 0.0, width = 0.8) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  theme_minimal() +
  theme(axis.text.x = element_blank()) +
  geom_point(data = subset(df, SampleID %in% selected_samples), aes(SampleID, value), shape = 21, fill = &quot;red&quot;, color = &quot;black&quot;, size = 1, show.legend = FALSE)</code></pre>
<p><img src="figure/009_quality_control.Rmd/absolute_counts_cutoff,%20-1.png" width="2700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-absolute_counts_cutoff,--1">
Past versions of "absolute_counts_cutoff, -1.png"
</button>
</p>
<div id="fig-no-spaces-absolute_counts_cutoff,--1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/009_quality_control.Rmd/absolute_counts_cutoff, -1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>## Plot relative
df %&gt;%
  ggplot(aes(SampleID, value, fill = Step)) +
  geom_bar(position = &quot;fill&quot;, stat = &quot;identity&quot;, alpha = 0.8, colour = 0.0) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  theme_minimal() +
  theme(axis.text.x = element_blank())</code></pre>
<p><img src="figure/009_quality_control.Rmd/relative_counts-1.png" width="2700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-relative_counts-1">
Past versions of relative_counts-1.png
</button>
</p>
<div id="fig-relative_counts-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/009_quality_control.Rmd/relative_counts-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here we can see the relative number of pair-ended reads (PE reads) of
each sample separated by QFail reads, QPass reads and Human reads.
Samples appear ordered depending on the total amount of Qpass reads.</p>
</div>
<div id="section" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>#&gt; R version 4.1.2 (2021-11-01)
#&gt; Platform: x86_64-pc-linux-gnu (64-bit)
#&gt; Running under: Ubuntu 22.04.4 LTS
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so
#&gt; 
#&gt; locale:
#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#&gt;  [3] LC_TIME=es_ES.UTF-8        LC_COLLATE=en_US.UTF-8    
#&gt;  [5] LC_MONETARY=es_ES.UTF-8    LC_MESSAGES=en_US.UTF-8   
#&gt;  [7] LC_PAPER=es_ES.UTF-8       LC_NAME=C                 
#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
#&gt; [11] LC_MEASUREMENT=es_ES.UTF-8 LC_IDENTIFICATION=C       
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     
#&gt; 
#&gt; other attached packages:
#&gt; [1] ggplot2_3.5.1  magrittr_2.0.3
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;   [1] utf8_1.2.4                  tidyselect_1.2.1           
#&gt;   [3] RSQLite_2.3.6               AnnotationDbi_1.56.2       
#&gt;   [5] htmlwidgets_1.6.4           grid_4.1.2                 
#&gt;   [7] BiocParallel_1.28.3         aws.signature_0.6.0        
#&gt;   [9] pROC_1.18.5                 munsell_0.5.1              
#&gt;  [11] codetools_0.2-20            effectsize_0.8.8           
#&gt;  [13] withr_3.0.0                 colorspace_2.1-0           
#&gt;  [15] Biobase_2.54.0              phyloseq_1.38.0            
#&gt;  [17] highr_0.10                  knitr_1.46                 
#&gt;  [19] rstudioapi_0.16.0           stats4_4.1.2               
#&gt;  [21] ggsignif_0.6.4              ggside_0.3.1               
#&gt;  [23] labeling_0.4.3              MatrixGenerics_1.6.0       
#&gt;  [25] git2r_0.33.0                GenomeInfoDbData_1.2.7     
#&gt;  [27] polyclip_1.10-6             bit64_4.0.5                
#&gt;  [29] farver_2.1.2                datawizard_0.10.0          
#&gt;  [31] rhdf5_2.38.1                rprojroot_2.0.4            
#&gt;  [33] vctrs_0.6.5                 generics_0.1.3             
#&gt;  [35] TH.data_1.1-2               xfun_0.44                  
#&gt;  [37] R6_2.5.1                    doParallel_1.0.17          
#&gt;  [39] GenomeInfoDb_1.30.1         clue_0.3-65                
#&gt;  [41] locfit_1.5-9.9              bitops_1.0-7               
#&gt;  [43] rhdf5filters_1.6.0          cachem_1.1.0               
#&gt;  [45] DelayedArray_0.20.0         promises_1.3.0             
#&gt;  [47] scales_1.3.0                multcomp_1.4-25            
#&gt;  [49] gtable_0.3.5                sandwich_3.1-0             
#&gt;  [51] workflowr_1.7.1             rlang_1.1.3                
#&gt;  [53] zeallot_0.1.0               genefilter_1.76.0          
#&gt;  [55] GlobalOptions_0.1.2         splines_4.1.2              
#&gt;  [57] rstatix_0.7.2               lazyeval_0.2.2             
#&gt;  [59] broom_1.0.6                 yaml_2.3.8                 
#&gt;  [61] reshape2_1.4.4              abind_1.4-5                
#&gt;  [63] backports_1.5.0             httpuv_1.6.15              
#&gt;  [65] ggmosaic_0.3.3              tools_4.1.2                
#&gt;  [67] jquerylib_0.1.4             biomformat_1.22.0          
#&gt;  [69] selbal_0.1.0                RColorBrewer_1.1-3         
#&gt;  [71] BiocGenerics_0.40.0         Rcpp_1.0.12                
#&gt;  [73] plyr_1.8.9                  base64enc_0.1-3            
#&gt;  [75] zlibbioc_1.40.0             purrr_1.0.2                
#&gt;  [77] RCurl_1.98-1.14             ggpubr_0.6.0               
#&gt;  [79] GetoptLong_1.0.5            correlation_0.8.4          
#&gt;  [81] S4Vectors_0.32.4            zoo_1.8-12                 
#&gt;  [83] SummarizedExperiment_1.24.0 ggrepel_0.9.5              
#&gt;  [85] cluster_2.1.6               fs_1.6.4                   
#&gt;  [87] here_1.0.1                  data.table_1.15.4          
#&gt;  [89] RSpectra_0.16-1             circlize_0.4.16            
#&gt;  [91] mvtnorm_1.2-5               whisker_0.4.1              
#&gt;  [93] matrixStats_1.3.0           hms_1.1.3                  
#&gt;  [95] patchwork_1.2.0             evaluate_0.23              
#&gt;  [97] xtable_1.8-4                XML_3.99-0.16.1            
#&gt;  [99] IRanges_2.28.0              gridExtra_2.3              
#&gt; [101] shape_1.4.6.1               compiler_4.1.2             
#&gt; [103] ellipse_0.5.0               tibble_3.2.1               
#&gt; [105] ggstatsplot_0.12.3          crayon_1.5.2               
#&gt; [107] htmltools_0.5.8.1           mgcv_1.9-1                 
#&gt; [109] corpcor_1.6.10              later_1.3.2                
#&gt; [111] tzdb_0.4.0                  geneplotter_1.72.0         
#&gt; [113] tidyr_1.3.1                 libcoin_1.0-10             
#&gt; [115] aws.s3_0.3.21               DBI_1.2.2                  
#&gt; [117] tweenr_2.0.3                ComplexHeatmap_2.18.0      
#&gt; [119] MASS_7.3-60.0.1             Matrix_1.3-4               
#&gt; [121] ade4_1.7-22                 car_3.1-2                  
#&gt; [123] readr_2.1.5                 permute_0.9-7              
#&gt; [125] cli_3.6.2                   parallel_4.1.2             
#&gt; [127] insight_0.19.11             igraph_2.0.3               
#&gt; [129] GenomicRanges_1.46.1        pkgconfig_2.0.3            
#&gt; [131] metar_0.1.5                 statsExpressions_1.5.4     
#&gt; [133] coin_1.4-3                  plotly_4.10.4              
#&gt; [135] xml2_1.3.6                  paletteer_1.6.0            
#&gt; [137] foreach_1.5.2               rARPACK_0.11-0             
#&gt; [139] annotate_1.72.0             bslib_0.7.0                
#&gt; [141] multtest_2.50.0             XVector_0.34.0             
#&gt; [143] stringr_1.5.1               digest_0.6.35              
#&gt; [145] parameters_0.21.7           vegan_2.6-6.1              
#&gt; [147] Biostrings_2.62.0           rmarkdown_2.27             
#&gt; [149] curl_5.2.1                  modeltools_0.2-23          
#&gt; [151] rjson_0.2.21                lifecycle_1.0.4            
#&gt; [153] nlme_3.1-164                jsonlite_1.8.8             
#&gt; [155] Rhdf5lib_1.16.0             carData_3.0-5              
#&gt; [157] mixOmics_6.18.1             viridisLite_0.4.2          
#&gt; [159] fansi_1.0.6                 pillar_1.9.0               
#&gt; [161] lattice_0.22-6              KEGGREST_1.34.0            
#&gt; [163] fastmap_1.2.0               httr_1.4.7                 
#&gt; [165] survival_3.6-4              glue_1.7.0                 
#&gt; [167] bayestestR_0.13.2           png_0.1-8                  
#&gt; [169] iterators_1.0.14            bit_4.0.5                  
#&gt; [171] ggforce_0.4.2               stringi_1.8.4              
#&gt; [173] sass_0.4.9                  blob_1.2.4                 
#&gt; [175] rematch2_2.1.2              DESeq2_1.34.0              
#&gt; [177] lefser_1.4.0                memoise_2.0.1              
#&gt; [179] dplyr_1.1.4                 ape_5.8</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
