<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Francesc Català-Moll &amp; Oriol Careta Borràs" />

<meta name="date" content="2024-06-04" />

<title>CHIR HIV RV306 Vaginal - Global Description</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">2024_cihr_hiv_rv306</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-check-square"></span>
     
    Quality control and initial microbiome exploration
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="004_throughput_analysis.html">Throughput Analysis</a>
    </li>
    <li>
      <a href="005_mre_initialization.html">Init mre object</a>
    </li>
    <li>
      <a href="007_phylosmith_abundance.html">Phylosmith abundance exploration</a>
    </li>
    <li>
      <a href="008_phylosmith_diversity.html">Phylosmith diversity exploration</a>
    </li>
    <li>
      <a href="009_quality_control.html">Quality_control</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-flag-checkered"></span>
     
    Analysis of baseline
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="100_global_description.html">Global description</a>
    </li>
    <li>
      <a href="101_baseline_timepoints.html">Baseline timepoints</a>
    </li>
    <li>
      <a href="102_starting_point_description.html">Starting point description</a>
    </li>
    <li>
      <a href="103_baseline_description.html">Baseline description</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-flask"></span>
     
    Analysis of vaccination effects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="104_groups_comparison.html">Groups Comparison</a>
    </li>
    <li>
      <a href="105_longitudinal_sudy_t1.html">Longitudinal Study of Group 1</a>
    </li>
    <li>
      <a href="105_longitudinal_sudy_t2.html">Longitudinal Study of Group 2</a>
    </li>
    <li>
      <a href="105_longitudinal_sudy_t3.html">Longitudinal Study of Group 3</a>
    </li>
    <li>
      <a href="105_longitudinal_sudy_t4A.html">Longitudinal Study of Group 4A</a>
    </li>
    <li>
      <a href="105_longitudinal_sudy_t4B.html">Longitudinal Study of Group 4B</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">CHIR HIV RV306 Vaginal - Global
Description</h1>
<h4 class="author">Francesc Català-Moll &amp; Oriol Careta Borràs</h4>
<h4 class="date">2024-06-04</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-06-04
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>OriolGEM.github.io/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240321code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240321)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240321code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240321)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomOriolGEMOriolGEMgithubiotree64c7378a0d3b7cd42b953f017d91ed4ba18aca01targetblank64c7378a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/tree/64c7378a0d3b7cd42b953f017d91ed4ba18aca01" target="_blank">64c7378</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomOriolGEMOriolGEMgithubiotree64c7378a0d3b7cd42b953f017d91ed4ba18aca01targetblank64c7378a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/OriolGEM/OriolGEM.github.io/tree/64c7378a0d3b7cd42b953f017d91ed4ba18aca01" target="_blank">64c7378</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/.RData
    Ignored:    code/.Rhistory
    Ignored:    data/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/100_global_description.Rmd</code>) and HTML
(<code>docs/100_global_description.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/2a85fa76788e4564afb6975d25032f9669ef98e2/analysis/100_global_description.Rmd" target="_blank">2a85fa7</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-04
</td>
<td>
updated images
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/OriolGEM/OriolGEM.github.io/2a85fa76788e4564afb6975d25032f9669ef98e2/docs/100_global_description.html" target="_blank">2a85fa7</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-04
</td>
<td>
updated images
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/analysis/100_global_description.Rmd" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/OriolGEM/OriolGEM.github.io/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/100_global_description.html" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>We are interested in a general description of the dataset in terms of
microbiome composition and function. We are using all available samples
(481) from all patients (118). The dataset is contains vaginal
microbiome data extracted 2 weeks after administering the vaccine
regimen at the corresponding study time-points, following the RV306
trial:</p>
<div class="float">
<img src="assets/Study.jpg" alt="Sudy Design" />
<div class="figcaption">Sudy Design</div>
</div>
</div>
<div id="load-mre" class="section level2">
<h2>Load mre</h2>
<pre class="r"><code>## mre init &amp; filter
bucket &lt;- &quot;s3://cihr-hiv-rv306/vaginal/metagenome/WMGS&quot;
out_dir &lt;- here::here(&quot;data/preprocessed/&quot;)
dir.create(out_dir, showWarnings = FALSE, recursive = TRUE)

mre &lt;- metar::metarExperiment_aws(bucket = bucket, wd = out_dir)

# Create new dataframe with the variable to add
df &lt;- mre@metadata@metadata_df
new_df &lt;- df %&gt;% 
  dplyr::select(SampleID, Treatment_group) %&gt;% 
  dplyr::mutate(
    Treatment_group_1 = dplyr::case_when(
      Treatment_group %in% &quot;t_4&quot; ~ &quot;t_4A&quot;,
      Treatment_group %in% &quot;t_no_boost&quot; ~ &quot;t_4A&quot;,
      TRUE ~ as.character (Treatment_group)
    )
  ) %&gt;% 
  dplyr::select(SampleID, Treatment_group_1)

# Add the new var to the mre
mre &lt;- metar::add_metadata_var(mre, new_df, id_col = &quot;SampleID&quot;)

# Add the new var as categorical var
mre &lt;- metar::add_cat(mre, &quot;Treatment_group_1&quot;, &quot;Set1&quot;)

mre &lt;- 
  metar::get_meta(mre) %&gt;% 
  dplyr::pull(SampleID) %&gt;% 
  metar::filter_samples(mre, sample_ids = .)

mre</code></pre>
<pre><code>#&gt; MetarSet-class experimental-level object 
#&gt;   [MetadataSet     ]: 481 samples and 7 variables 
#&gt;   [DiversitySet    ]: 
#&gt;      ✖ igc: 0 observations and 0 samples 
#&gt;      ✔ virgo: 14765 observations and 481 samples 
#&gt;      ✖ dada2: 0 observations and 0 samples 
#&gt;   [TaxaSet         ]: 
#&gt;      ✖ metaphlan: no phyloseq object available 
#&gt;      ✖ kraken: no phyloseq object available 
#&gt;      ✖ bracken: no phyloseq object available 
#&gt;      ✖ motus: no phyloseq object available 
#&gt;      ✔ virgo: phyloseq object with 258 taxa and 481 samples 
#&gt;      ✖ dada2: no phyloseq object available 
#&gt;   [GeneFunctionSet ]: 
#&gt;      ✖ humann: no functional data aviable 
#&gt;      ✖ igc: no functional data aviable 
#&gt;      ✔ virgo: 4621 kegg_id, 0 cog_id, 0 eggnog_id, 0 ec_id and 0 metacyc_id 
#&gt;      ✖ fmap: no functional data aviable 
#&gt;      ✖ picrust2: no functional data aviable 
#&gt;      ✖ tax4fun: no functional data aviable 
#&gt;      ✖ phylolisft: no functional data aviable</code></pre>
</div>
<div id="qc" class="section level2">
<h2>QC</h2>
<p>We will perform a brief QC as we have already performed it.</p>
<pre class="r"><code>## def vars
bucket &lt;- &quot;s3://cihr-hiv-rv306&quot;
filt_data_obj &lt;- &quot;vaginal/metagenome/WMGS/RawData/DataFiltering_Boxplot.rds&quot;

## load seq analysis data
df &lt;- 
  aws.s3::s3readRDS(filt_data_obj, bucket) %&gt;% 
  .[[1]] %&gt;% ## Get data tibble from ggplot2 object
  tibble::as_tibble() %&gt;% 
  dplyr::filter(SampleID %in% metar::get_meta(mre)$SampleID)

## Plot absolute
df %&gt;%
  ggplot(aes(SampleID, value, fill = Step)) +
  geom_bar(position = &quot;stack&quot;, stat = &quot;identity&quot;, alpha = 0.8, colour = 0.0, width = 0.8) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  theme_minimal() +
  theme(axis.text.x = element_blank())</code></pre>
<p><img src="figure/100_global_description.Rmd/absolute_counts-1.png" width="2700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-absolute_counts-1">
Past versions of absolute_counts-1.png
</button>
</p>
<div id="fig-absolute_counts-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/absolute_counts-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here we can see the total number of pair-ended reads (PE reads) of
each sample separated by QFail reads, QPass reads and Human reads.
Samples appear ordered depending on the amount of Qpass reads. Thus,
here we cannot only see the amount of QPass reads of each sample but
also the sampling depth of each sample.</p>
<pre class="r"><code>## Plot relative
df %&gt;%
  ggplot(aes(SampleID, value, fill = Step)) +
  geom_bar(position = &quot;fill&quot;, stat = &quot;identity&quot;, alpha = 0.8, colour = 0.0) +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  theme_minimal() +
  theme(axis.text.x = element_blank())</code></pre>
<p><img src="figure/100_global_description.Rmd/relative_counts-1.png" width="2700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-relative_counts-1">
Past versions of relative_counts-1.png
</button>
</p>
<div id="fig-relative_counts-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/relative_counts-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here we can see the relative number of pair-ended reads (PE reads) of
each sample separated by QFail reads, QPass reads and Human reads.
Samples appear ordered depending on the total amount of Qpass reads. As
it is relativized, we are not able to see sample depth but we can
compare the percentage of QPass reads from the total amount on reads
between all samples.</p>
<p>We will also add a categorical variable where we distribute the QPass
reads in quantiles, as usually a higher number of QPass reads means
higher diversity observed, but we want to check if it is be the
case.</p>
<pre class="r"><code>## Add new metadata var
mre &lt;- 
  dplyr::filter(df, Step == &quot;QPassReads&quot;) %&gt;% 
  dplyr::mutate(
    pass_reads_quantile = findInterval(value, quantile(value)),
    pass_reads_quantile = dplyr::case_when(
      pass_reads_quantile == 5 ~ &quot;c_4&quot;, 
      TRUE ~ stringr::str_c(&quot;c_&quot;, pass_reads_quantile)
    )
  ) %&gt;% 
  dplyr::select(SampleID, pass_reads_quantile) %&gt;% 
  metar::add_metadata_var(mre, .) %&gt;% 
  metar::add_cat(&quot;pass_reads_quantile&quot;, &quot;Pastel2&quot;)</code></pre>
</div>
<div id="section" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
</div>
<div id="alpha-diversity---rarefaction" class="section level2 tabset">
<h2 class="tabset">Alpha Diversity - Rarefaction</h2>
<p>Note that diversity analysis here are delicate to perform due to the
limitations in sampling.</p>
<p>From the taxonomic composition we’ll derive some alpha-diversity
ecological indices and compare them among groups.</p>
<p>We also want to use the VIRGO catalog mapping-derived gene richness.
However, this can be problematic for gene richness since the initial
reads per sample is very low in some cases. This low numbers may be
enough to describe the taxonomic composition, specially if the
microbiome is “simple”, but may fail to capture richness/diversity.</p>
<pre class="r"><code>## run rarefaction
mre &lt;- metar::virgo_rarefaction(mre, save_files = FALSE)

metar::get_cat(mre) %&gt;%
  dplyr::pull(1) %&gt;%
  .[-1] %&gt;% # Remove &quot;Treatment_group&quot;
  purrr::walk(~ {
    cat(&#39;\n\n### `&#39;, .x, &#39;`\n\n&#39;)
    metar::get_diversity(mre, type = &quot;virgo&quot;, res_slot = &quot;rarefaction&quot;) %&gt;%
      purrr::pluck(&quot;categorical&quot;, .x, &quot;rarefaction&quot;) %&gt;%
      plot()
  })</code></pre>
<div id="arm" class="section level3">
<h3><code>Arm</code></h3>
<p><img src="figure/100_global_description.Rmd/alpha_d_rarefraction-1.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_rarefraction-1">
Past versions of alpha_d_rarefraction-1.png
</button>
</p>
<div id="fig-alpha_d_rarefraction-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_rarefraction-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="heme_test" class="section level3">
<h3><code>Heme_test</code></h3>
<p><img src="figure/100_global_description.Rmd/alpha_d_rarefraction-2.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_rarefraction-2">
Past versions of alpha_d_rarefraction-2.png
</button>
</p>
<div id="fig-alpha_d_rarefraction-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_rarefraction-2.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="medical_center" class="section level3">
<h3><code>Medical_center</code></h3>
<p><img src="figure/100_global_description.Rmd/alpha_d_rarefraction-3.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_rarefraction-3">
Past versions of alpha_d_rarefraction-3.png
</button>
</p>
<div id="fig-alpha_d_rarefraction-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_rarefraction-3.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="treatment_group_1" class="section level3">
<h3><code>Treatment_group_1</code></h3>
<p><img src="figure/100_global_description.Rmd/alpha_d_rarefraction-4.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_rarefraction-4">
Past versions of alpha_d_rarefraction-4.png
</button>
</p>
<div id="fig-alpha_d_rarefraction-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_rarefraction-4.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pass_reads_quantile" class="section level3">
<h3><code>pass_reads_quantile</code></h3>
<p><img src="figure/100_global_description.Rmd/alpha_d_rarefraction-5.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_rarefraction-5">
Past versions of alpha_d_rarefraction-5.png
</button>
</p>
<div id="fig-alpha_d_rarefraction-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_rarefraction-5.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="section-1" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
<p>The rarefaction curve shows how far we are from assymptotically
saturating richness signal from this catalog at the sampling depth we
have (after filtering human sequences). Thus, we need to be aware of the
limitations of gene richness analysis and interpret with caution. When
comparing between variable groups, sampling depth seem to be no
different between them, except, logically, for the variable Qpass reads
quantile, where, depending on the quantile, they have more or less
sample depth.</p>
</div>
<div id="alpha-diversity---boxplots" class="section level2 tabset">
<h2 class="tabset">Alpha Diversity - Boxplots</h2>
<pre class="r"><code>## plot gene richness
metar::get_cat(mre) %&gt;%
  dplyr::pull(1) %&gt;%
  .[-1] %&gt;% # Remove &quot;Treatment_group&quot;
  purrr::walk(~ {
    cat(&#39;\n\n### `&#39;, .x, &#39;`\n\n&#39;)
    metar::get_diversity(mre, type = &quot;virgo&quot;, res_slot = &quot;rarefaction&quot;) %&gt;%
      purrr::pluck(&quot;categorical&quot;, .x, &quot;boxplots&quot;, &quot;GeneNumber&quot;) %&gt;%
      plot()
  })</code></pre>
<div id="arm-1" class="section level3">
<h3><code>Arm</code></h3>
<p><img src="figure/100_global_description.Rmd/alpha_d_boxplots-1.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_boxplots-1">
Past versions of alpha_d_boxplots-1.png
</button>
</p>
<div id="fig-alpha_d_boxplots-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_boxplots-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="heme_test-1" class="section level3">
<h3><code>Heme_test</code></h3>
<p><img src="figure/100_global_description.Rmd/alpha_d_boxplots-2.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_boxplots-2">
Past versions of alpha_d_boxplots-2.png
</button>
</p>
<div id="fig-alpha_d_boxplots-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_boxplots-2.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="medical_center-1" class="section level3">
<h3><code>Medical_center</code></h3>
<p><img src="figure/100_global_description.Rmd/alpha_d_boxplots-3.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_boxplots-3">
Past versions of alpha_d_boxplots-3.png
</button>
</p>
<div id="fig-alpha_d_boxplots-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_boxplots-3.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="treatment_group_1-1" class="section level3">
<h3><code>Treatment_group_1</code></h3>
<p><img src="figure/100_global_description.Rmd/alpha_d_boxplots-4.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_boxplots-4">
Past versions of alpha_d_boxplots-4.png
</button>
</p>
<div id="fig-alpha_d_boxplots-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_boxplots-4.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pass_reads_quantile-1" class="section level3">
<h3><code>pass_reads_quantile</code></h3>
<p><img src="figure/100_global_description.Rmd/alpha_d_boxplots-5.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_boxplots-5">
Past versions of alpha_d_boxplots-5.png
</button>
</p>
<div id="fig-alpha_d_boxplots-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_boxplots-5.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="section-2" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
<p>The gene richness boxplots indicate that there are no significant
differences between groups of variables in terms of gene richness,
except when separating by treatment group, where the gene richness of
t_2 is significantly higher compared to t_4A. There are also significant
differences between gene richness when observing by Qpass reads
quantiles, as a higher number of Qpass reads (higher sampling depth)
usually leads to a higher gene richness.</p>
</div>
<div id="alpha-diversity---longitudinal" class="section level2 tabset">
<h2 class="tabset">Alpha Diversity - Longitudinal</h2>
<p>To further explore the aforementioned observations, we will verify
whether gene diversity varies across time-points for the categorical
variables.</p>
<pre class="r"><code>metar::get_cat(mre) %&gt;%
  dplyr::pull(1) %&gt;%
  .[-1] %&gt;% # Remove &quot;Treatment_group&quot;
  purrr::walk(~ { 
    cat(&#39;\n\n### `&#39;, .x, &#39;`\n\n&#39;)
    metar::get_diversity(mre, type = &quot;virgo&quot;, res_slot = &quot;rarefaction&quot;) %&gt;%
      purrr::pluck(&quot;longitudinal_rel&quot;, &quot;categorical&quot;, .x) %&gt;%
      plot()
  })</code></pre>
<div id="arm-2" class="section level3">
<h3><code>Arm</code></h3>
<p><img src="figure/100_global_description.Rmd/lon_gene_richness-1.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-lon_gene_richness-1">
Past versions of lon_gene_richness-1.png
</button>
</p>
<div id="fig-lon_gene_richness-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/lon_gene_richness-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="heme_test-2" class="section level3">
<h3><code>Heme_test</code></h3>
<p><img src="figure/100_global_description.Rmd/lon_gene_richness-2.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-lon_gene_richness-2">
Past versions of lon_gene_richness-2.png
</button>
</p>
<div id="fig-lon_gene_richness-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/lon_gene_richness-2.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="medical_center-2" class="section level3">
<h3><code>Medical_center</code></h3>
<p><img src="figure/100_global_description.Rmd/lon_gene_richness-3.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-lon_gene_richness-3">
Past versions of lon_gene_richness-3.png
</button>
</p>
<div id="fig-lon_gene_richness-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/lon_gene_richness-3.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="treatment_group_1-2" class="section level3">
<h3><code>Treatment_group_1</code></h3>
<p><img src="figure/100_global_description.Rmd/lon_gene_richness-4.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-lon_gene_richness-4">
Past versions of lon_gene_richness-4.png
</button>
</p>
<div id="fig-lon_gene_richness-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/lon_gene_richness-4.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pass_reads_quantile-2" class="section level3">
<h3><code>pass_reads_quantile</code></h3>
<p><img src="figure/100_global_description.Rmd/lon_gene_richness-5.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-lon_gene_richness-5">
Past versions of lon_gene_richness-5.png
</button>
</p>
<div id="fig-lon_gene_richness-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/lon_gene_richness-5.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="section-3" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
<p>In these graphs, we can see the changes in the relative gene richness
over time of samples separated by categorical variable groups. It seems
that while gene richness is stable on the active arm group over time,
gene richness on the placebo group is more variable. Regarding the other
variables, gene richness seems constant over time.</p>
</div>
<div id="beta-diversity---species-composition" class="section level2">
<h2>Beta-Diversity - Species Composition</h2>
<p>Let’s visualize composition at the species level using barplots.</p>
<pre class="r"><code>## run barplots 
mre &lt;- metar::virgo_barplots(mre, top_n = 50, save_files = FALSE)

## Plot virgo barplots
mre %&gt;%
  metar::get_taxa(&quot;virgo&quot;, &quot;barplots&quot;) %&gt;%
  purrr::pluck(&quot;ta1&quot;, &quot;top_50&quot;, &quot;rel_abundance&quot;, &quot;bray_hclust_order_barplot&quot;)</code></pre>
<p><img src="figure/100_global_description.Rmd/abundance_barplots-1.png" width="3600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-abundance_barplots-1">
Past versions of abundance_barplots-1.png
</button>
</p>
<div id="fig-abundance_barplots-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/abundance_barplots-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The relative abundance barplots in hierarchical clustering order
depicting taxonomic diversity per sample reveal the presence of three
distinct groups:</p>
<ul>
<li>The first group is primarily characterized by the dominance of
<strong>Gardnerella vaginalis</strong>, occasionally accompanied by
co-dominance with <strong>Atopobium vaginae</strong>. Notably, this
group exhibits a higher level of taxonomic diversity compared to the
other two groups. Both <strong>Gardnerella vaginalis</strong>, a
gram-negative facultative anaerobic bacterium, and <strong>Atopobium
vaginae</strong>, a gram-positive strict anaerobic bacterium, have been
associated with bacterial vaginosis (BV) and its characteristic
symptoms.</li>
<li>The second group, is characterized by a strong dominance of
<strong>Lactobacillus iners</strong>. <strong>Lactobacillus
iners</strong> is a common bacterium found in the vaginal microbiota,
playing a vital role in maintaining vaginal health. It helps to maintain
an acidic pH, creating an unfavorable environment for harmful
microorganisms. <strong>Lactobacillus iners</strong> contributes to the
stability and balance of the vaginal microbiota, promoting overall
vaginal health and potentially protecting against infections like
bacterial vaginosis.</li>
<li>The third group is mainly characterized by a dominant presence of
<strong>Lactobacillus crispatus</strong>, although some samples in this
group also show high levels of <strong>Lactobacillus iners</strong>.
<strong>Lactobacillus crispatus</strong> is a beneficial bacterium
commonly found in the vaginal microbiota. It helps maintain a healthy
vaginal environment by producing lactic acid and supporting
acidity.</li>
</ul>
<pre class="r"><code>## Count of two predominant taxas per sample
mre %&gt;% 
  metar::get_taxa(&quot;virgo&quot;, &quot;barplots&quot;) %&gt;%
  purrr::pluck(&quot;ta1&quot;, &quot;top_50&quot;, &quot;rel_abundance&quot;, &quot;bray_hclust_order_barplot&quot;, &quot;data&quot;) %&gt;%
  dplyr::group_by(SampleID) %&gt;%
  dplyr::slice_max(Abundance, n = 2) %&gt;%
  dplyr::ungroup() %&gt;%
  dplyr::count(Species, sort = TRUE)</code></pre>
<pre><code>#&gt; # A tibble: 23 × 2
#&gt;    Species                      n
#&gt;    &lt;fct&gt;                    &lt;int&gt;
#&gt;  1 Gardnerella_vaginalis      246
#&gt;  2 Lactobacillus_iners        238
#&gt;  3 Lactobacillus_crispatus    192
#&gt;  4 Atopobium_vaginae          120
#&gt;  5 Lactobacillus_jensenii      79
#&gt;  6 Lactobacillus_gasseri       16
#&gt;  7 BVAB1                       14
#&gt;  8 Prevotella_bivia            13
#&gt;  9 Prevotella_amnii             9
#&gt; 10 Streptococcus_agalactiae     7
#&gt; # ℹ 13 more rows</code></pre>
<p>To confirm these observations, we created a table where we counted
the two most dominant species per each sample. It can be seen that the
most abundant species are <strong>Gardnerella vaginalis</strong>,
<strong>Lactobacillus iners</strong> and <strong>Lactobacillus
crispatus</strong>, followed by <strong>Atopobium vaginae</strong>,
which on the abundance barplot was found in codominance with
<strong>Gardnerella vaginalis</strong> on some samples. The fourth and
fifth more common species are also <strong>Lactobacillus</strong>.</p>
</div>
<div id="beta-diversity---cluestering" class="section level2">
<h2>Beta-Diversity - Cluestering</h2>
<p>In the previous barplots, the x-axis is defined according to ward.D2,
bray-based clustering. We’ll now transform abundances and project them
into a heatmap, to be able to visualize association with variables of
interest.</p>
<pre class="r"><code>mre &lt;- metar::virgo_heatmap(mre, top_n = 50, save_files = FALSE)

mre %&gt;% 
  metar::get_taxa(&quot;virgo&quot;, &quot;heatmaps&quot;) %&gt;%
  purrr::pluck(&quot;ta1&quot;, &quot;top_50&quot;) %&gt;%
  ComplexHeatmap::draw(heatmap_legend_side = &quot;left&quot;, annotation_legend_side = &quot;bottom&quot;)</code></pre>
<p><img src="figure/100_global_description.Rmd/clust_heatmap-1.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-clust_heatmap-1">
Past versions of clust_heatmap-1.png
</button>
</p>
<div id="fig-clust_heatmap-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/clust_heatmap-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>In this heatmap with hierarchical clustering, we can clearly see the
three groups, but there is no apparent association between composition
and metadata variables. In some samples dominated by <strong>Gardnerella
vaginalis</strong>, there is co-dominance with <strong>Atopobium
vaginae</strong> present.</p>
</div>
<div id="beta-diversity---unsupervised-cluestering"
class="section level2">
<h2>Beta-Diversity - Unsupervised Cluestering</h2>
<p>We have seen that the vaginal microbiome composition is quite
discretized in many of the samples. That is, there is a single species
which dominate in most of the samples, while the rest have a combination
fo the rest of the species, with higher dominance indices (we can check
that separately). So, the question is whether there are composition
types (from now on, called vaginotypes) that we can detect.</p>
<p>For consistence with the rest of the analysis, we’ll perform the PAM
(Partitioning Around Medoids) clustering analysis using all samples in
order to broadly detect these vagintoypes and be able to uniquely assign
each sample to a vaginotype, while ensuring that this assignment does
not changes in downstream analysis.</p>
<p>The PAM algorithm searches for k representative objects in a data set
(k medoids) and then assigns each object to the closest medoid in order
to create clusters. Its aim is to minimize the sum of dissimilarities
between the objects in a cluster and the center of the same cluster
(medoid).</p>
<pre class="r"><code>## run nmds
mre &lt;- metar::virgo_nmds(mre, top_n = 50, save_files = FALSE)

## Silhouette plot
mre %&gt;% 
  metar::get_taxa(&quot;virgo&quot;, &quot;nmds&quot;) %&gt;%
  purrr::pluck(&quot;ta1&quot;, &quot;top_50&quot;, &quot;clustering_results&quot;, &quot;silhouette_plt&quot;)</code></pre>
<p><img src="figure/100_global_description.Rmd/clusters-1.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-clusters-1">
Past versions of clusters-1.png
</button>
</p>
<div id="fig-clusters-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/clusters-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The Silhouette plot shows that the silhouette coefficient is highest
when k=3, suggesting that’s the optimal number of clusters. This may be
pointing to 3 different vaginotypes in the entire sample set. Moreover,
it agrees with what was observed in the barplot and the heatmap.</p>
<pre class="r"><code>## Cluster summary
mre %&gt;% 
  metar::get_taxa(&quot;virgo&quot;, &quot;nmds&quot;) %&gt;%
  purrr::pluck(&quot;ta1&quot;, &quot;top_50&quot;, &quot;clustering_results&quot;, &quot;sample_cluster_labels&quot;) %&gt;%
  dplyr::count(cluster)</code></pre>
<pre><code>#&gt; # A tibble: 3 × 2
#&gt;   cluster     n
#&gt;     &lt;int&gt; &lt;int&gt;
#&gt; 1       1   115
#&gt; 2       2   177
#&gt; 3       3   189</code></pre>
<p>This table shows us the number of samples that are classified into
each of these 3 clusters.</p>
<p>Nevertheless, we are interested in the main characteristics of these
vaginotypes, so we will use the cluster label to plot a heatmap and
visualize main characteristics.</p>
<pre class="r"><code>## Cluster nmds
mre %&gt;% 
  metar::get_taxa(&quot;virgo&quot;, &quot;nmds&quot;) %&gt;%
  purrr::pluck(&quot;ta1&quot;, &quot;top_50&quot;, &quot;categorical&quot;, &quot;cluster&quot;)</code></pre>
<p><img src="figure/100_global_description.Rmd/cluster_nmds-1.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cluster_nmds-1">
Past versions of cluster_nmds-1.png
</button>
</p>
<div id="fig-cluster_nmds-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/cluster_nmds-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Apparently:</p>
<ul>
<li><p>Cluster 1: Is enriched with <strong>L. crispatus</strong>,
although some samples in this group also show high levels of <strong>L.
iners</strong> and other Lactobacillus.</p></li>
<li><p>Cluster 2: Is mainly <strong>G. vaginalis</strong>, occasionally
accompanied by co-dominance with <strong>A. vaginae</strong>.</p></li>
<li><p>Cluster 3: Is dominated by <strong>L. iners</strong>.</p></li>
</ul>
<p>We will first create a variable for these clusters, and we also added
a categorical variable based on whether the vaginotype has pathological
potential or not. We call the categorical variable <strong>vaginotype
condition</strong> which includes the categories <strong>healty</strong>
and <strong>vaginosis</strong>.</p>
<pre class="r"><code>## Prepare var
vaginotypes &lt;- 
  mre %&gt;% 
  metar::get_taxa(&quot;virgo&quot;, &quot;nmds&quot;) %&gt;% 
  purrr::pluck(&quot;ta1&quot;, &quot;top_50&quot;, &quot;clustering_results&quot;, &quot;sample_cluster_labels&quot;) %&gt;% 
  dplyr::mutate(
    vaginotype = dplyr::case_when(
      cluster == 1 ~ &quot;c1_L_crispatus&quot;, 
      cluster == 2 ~ &quot;c2_G_vaginalis&quot;, 
      cluster == 3 ~ &quot;c3_L_iners&quot;
    ), 
    vaginotype_condition = dplyr::if_else(cluster == 2, &quot;vaginosis&quot;, &quot;no_vaginosis&quot;)
  ) %&gt;% 
  dplyr::select(SampleID, vaginotype, vaginotype_condition)

## Update mre metadata
mre &lt;- 
  metar::add_metadata_var(mre, vaginotypes) %&gt;% 
  metar::add_cat(var = &quot;vaginotype&quot;, palette = &quot;Accent&quot;) %&gt;% 
  metar::add_cat(var = &quot;vaginotype_condition&quot;, palette = &quot;Set1&quot;)

## Save mre with vaginotype metadata variable
readr::write_rds(mre, here::here(&quot;data&quot;, &quot;preprocessed&quot;, &quot;mre_vaginotype.rds&quot;))</code></pre>
<p>If we take a closer look to clusters, in terms of Alpha
Diversity:</p>
</div>
<div id="alpha-diversity---rarefraction" class="section level2 tabset">
<h2 class="tabset">Alpha Diversity - Rarefraction</h2>
<pre class="r"><code>## Run rarefaction for new var
mre &lt;- metar::virgo_rarefaction(mre, save_files = FALSE)

metar::get_cat(mre) %&gt;%
  dplyr::pull(1) %&gt;%
  .[-1] %&gt;% # Remove &quot;Treatment_group&quot;
  purrr::walk(~ {
    cat(&#39;\n\n### `&#39;, .x, &#39;`\n\n&#39;)
    metar::get_diversity(mre, type = &quot;virgo&quot;, res_slot = &quot;rarefaction&quot;) %&gt;%
      purrr::pluck(&quot;categorical&quot;, .x, &quot;rarefaction&quot;) %&gt;%
      plot()
  })</code></pre>
<div id="arm-3" class="section level3">
<h3><code>Arm</code></h3>
<p><img src="figure/100_global_description.Rmd/alpha_d_rarefraction_2-1.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_rarefraction_2-1">
Past versions of alpha_d_rarefraction_2-1.png
</button>
</p>
<div id="fig-alpha_d_rarefraction_2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_rarefraction_2-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="heme_test-3" class="section level3">
<h3><code>Heme_test</code></h3>
<p><img src="figure/100_global_description.Rmd/alpha_d_rarefraction_2-2.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_rarefraction_2-2">
Past versions of alpha_d_rarefraction_2-2.png
</button>
</p>
<div id="fig-alpha_d_rarefraction_2-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_rarefraction_2-2.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="medical_center-3" class="section level3">
<h3><code>Medical_center</code></h3>
<p><img src="figure/100_global_description.Rmd/alpha_d_rarefraction_2-3.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_rarefraction_2-3">
Past versions of alpha_d_rarefraction_2-3.png
</button>
</p>
<div id="fig-alpha_d_rarefraction_2-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_rarefraction_2-3.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="treatment_group_1-3" class="section level3">
<h3><code>Treatment_group_1</code></h3>
<p><img src="figure/100_global_description.Rmd/alpha_d_rarefraction_2-4.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_rarefraction_2-4">
Past versions of alpha_d_rarefraction_2-4.png
</button>
</p>
<div id="fig-alpha_d_rarefraction_2-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_rarefraction_2-4.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pass_reads_quantile-3" class="section level3">
<h3><code>pass_reads_quantile</code></h3>
<p><img src="figure/100_global_description.Rmd/alpha_d_rarefraction_2-5.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_rarefraction_2-5">
Past versions of alpha_d_rarefraction_2-5.png
</button>
</p>
<div id="fig-alpha_d_rarefraction_2-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_rarefraction_2-5.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="vaginotype" class="section level3">
<h3><code>vaginotype</code></h3>
<p><img src="figure/100_global_description.Rmd/alpha_d_rarefraction_2-6.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_rarefraction_2-6">
Past versions of alpha_d_rarefraction_2-6.png
</button>
</p>
<div id="fig-alpha_d_rarefraction_2-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_rarefraction_2-6.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="vaginotype_condition" class="section level3">
<h3><code>vaginotype_condition</code></h3>
<p><img src="figure/100_global_description.Rmd/alpha_d_rarefraction_2-7.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_rarefraction_2-7">
Past versions of alpha_d_rarefraction_2-7.png
</button>
</p>
<div id="fig-alpha_d_rarefraction_2-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_rarefraction_2-7.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="section-4" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
<p>The rarefaction curve shows that when adding the categorical variable
vaginotype, the samples which are more rich in <strong>G.
Vaginalis</strong> are closer to asymptotically saturating richness
signal from this catalog, mainly because they tend to have a higher
sampling depth.</p>
</div>
<div id="alpha-diversity---boxplots-1" class="section level2 tabset">
<h2 class="tabset">Alpha Diversity - Boxplots</h2>
<pre class="r"><code>## Plotting
metar::get_cat(mre) %&gt;%
  dplyr::pull(1) %&gt;%
  .[-1] %&gt;% # Remove &quot;Treatment_group&quot;
  purrr::walk(~ {
    cat(&#39;\n\n### `&#39;, .x, &#39;`\n\n&#39;)
    metar::get_diversity(mre, type = &quot;virgo&quot;, res_slot = &quot;rarefaction&quot;) %&gt;%
      purrr::pluck(&quot;categorical&quot;, .x, &quot;boxplots&quot;, &quot;GeneNumber&quot;) %&gt;%
      plot()
  })</code></pre>
<pre><code>#&gt; 
#&gt; 
#&gt; ### ` Arm `</code></pre>
<p><img src="figure/100_global_description.Rmd/alpha_d_boxplots_2-1.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_boxplots_2-1">
Past versions of alpha_d_boxplots_2-1.png
</button>
</p>
<div id="fig-alpha_d_boxplots_2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_boxplots_2-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>#&gt; 
#&gt; 
#&gt; ### ` Heme_test `</code></pre>
<p><img src="figure/100_global_description.Rmd/alpha_d_boxplots_2-2.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_boxplots_2-2">
Past versions of alpha_d_boxplots_2-2.png
</button>
</p>
<div id="fig-alpha_d_boxplots_2-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_boxplots_2-2.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>#&gt; 
#&gt; 
#&gt; ### ` Medical_center `</code></pre>
<p><img src="figure/100_global_description.Rmd/alpha_d_boxplots_2-3.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_boxplots_2-3">
Past versions of alpha_d_boxplots_2-3.png
</button>
</p>
<div id="fig-alpha_d_boxplots_2-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_boxplots_2-3.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>#&gt; 
#&gt; 
#&gt; ### ` Treatment_group_1 `</code></pre>
<p><img src="figure/100_global_description.Rmd/alpha_d_boxplots_2-4.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_boxplots_2-4">
Past versions of alpha_d_boxplots_2-4.png
</button>
</p>
<div id="fig-alpha_d_boxplots_2-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_boxplots_2-4.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>#&gt; 
#&gt; 
#&gt; ### ` pass_reads_quantile `</code></pre>
<p><img src="figure/100_global_description.Rmd/alpha_d_boxplots_2-5.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_boxplots_2-5">
Past versions of alpha_d_boxplots_2-5.png
</button>
</p>
<div id="fig-alpha_d_boxplots_2-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_boxplots_2-5.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>#&gt; 
#&gt; 
#&gt; ### ` vaginotype `</code></pre>
<p><img src="figure/100_global_description.Rmd/alpha_d_boxplots_2-6.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_boxplots_2-6">
Past versions of alpha_d_boxplots_2-6.png
</button>
</p>
<div id="fig-alpha_d_boxplots_2-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_boxplots_2-6.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>#&gt; 
#&gt; 
#&gt; ### ` vaginotype_condition `</code></pre>
<p><img src="figure/100_global_description.Rmd/alpha_d_boxplots_2-7.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha_d_boxplots_2-7">
Past versions of alpha_d_boxplots_2-7.png
</button>
</p>
<div id="fig-alpha_d_boxplots_2-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alpha_d_boxplots_2-7.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="section-5" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
<p>After representing the gene richness for each level of each
categorical variable, we observed that there are also significant
differences within the categorical variables vaginotype. We noticed that
Cluster 2 (dominated by <strong>G. Vaginalis</strong> and potentially
associated with vaginosis) exhibits higher gene diversity compared to
the other two clusters. Furthermore, it is worth mentioning that the
other two clusters also display variations in gene diversity among
themselves.</p>
</div>
<div id="alpha-diversity---longitudinal-1"
class="section level2 tabset">
<h2 class="tabset">Alpha Diversity - Longitudinal</h2>
<p>To further explore the aforementioned observations, we will verify
whether gene diversity varies across timepoints for the categorical
variables.</p>
<pre class="r"><code>metar::get_cat(mre) %&gt;%
  dplyr::pull(1) %&gt;%
  .[-1] %&gt;% # Remove &quot;Treatment_group&quot;
  purrr::walk(~ { 
    cat(&#39;\n\n### `&#39;, .x, &#39;`\n\n&#39;)
    metar::get_diversity(mre, type = &quot;virgo&quot;, res_slot = &quot;rarefaction&quot;) %&gt;%
      purrr::pluck(&quot;longitudinal_rel&quot;, &quot;categorical&quot;, .x) %&gt;%
      plot()
  })</code></pre>
<div id="arm-4" class="section level3">
<h3><code>Arm</code></h3>
<p><img src="figure/100_global_description.Rmd/lon_gene_richness_2-1.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-lon_gene_richness_2-1">
Past versions of lon_gene_richness_2-1.png
</button>
</p>
<div id="fig-lon_gene_richness_2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/lon_gene_richness_2-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="heme_test-4" class="section level3">
<h3><code>Heme_test</code></h3>
<p><img src="figure/100_global_description.Rmd/lon_gene_richness_2-2.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-lon_gene_richness_2-2">
Past versions of lon_gene_richness_2-2.png
</button>
</p>
<div id="fig-lon_gene_richness_2-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/lon_gene_richness_2-2.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="medical_center-4" class="section level3">
<h3><code>Medical_center</code></h3>
<p><img src="figure/100_global_description.Rmd/lon_gene_richness_2-3.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-lon_gene_richness_2-3">
Past versions of lon_gene_richness_2-3.png
</button>
</p>
<div id="fig-lon_gene_richness_2-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/lon_gene_richness_2-3.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="treatment_group_1-4" class="section level3">
<h3><code>Treatment_group_1</code></h3>
<p><img src="figure/100_global_description.Rmd/lon_gene_richness_2-4.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-lon_gene_richness_2-4">
Past versions of lon_gene_richness_2-4.png
</button>
</p>
<div id="fig-lon_gene_richness_2-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/lon_gene_richness_2-4.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pass_reads_quantile-4" class="section level3">
<h3><code>pass_reads_quantile</code></h3>
<p><img src="figure/100_global_description.Rmd/lon_gene_richness_2-5.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-lon_gene_richness_2-5">
Past versions of lon_gene_richness_2-5.png
</button>
</p>
<div id="fig-lon_gene_richness_2-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/lon_gene_richness_2-5.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="vaginotype-1" class="section level3">
<h3><code>vaginotype</code></h3>
<p><img src="figure/100_global_description.Rmd/lon_gene_richness_2-6.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-lon_gene_richness_2-6">
Past versions of lon_gene_richness_2-6.png
</button>
</p>
<div id="fig-lon_gene_richness_2-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/lon_gene_richness_2-6.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="vaginotype_condition-1" class="section level3">
<h3><code>vaginotype_condition</code></h3>
<p><img src="figure/100_global_description.Rmd/lon_gene_richness_2-7.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-lon_gene_richness_2-7">
Past versions of lon_gene_richness_2-7.png
</button>
</p>
<div id="fig-lon_gene_richness_2-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/lon_gene_richness_2-7.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="section-6" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
<p>In these graphs, we can see the changes in the relative gene richness
over time of samples separated by categorical variable groups. It seems
that the cluster dominated by <strong>L. Iners</strong> has a less
stable microbiome over time.</p>
</div>
<div id="beta-diversity---cluestering-heatmap" class="section level2">
<h2>Beta-Diversity - Cluestering Heatmap</h2>
<p>In order to visualize how the newly generated categorical variables
cluster, we will once again represent the Heatmap for the 50 most
abundant bacterial species.</p>
<pre class="r"><code>## run heatmap
mre &lt;- metar::virgo_heatmap(mre, top_n = 50, save_files = FALSE)

## plot heatmap
mre %&gt;% 
  metar::get_taxa(&quot;virgo&quot;, &quot;heatmaps&quot;) %&gt;%
  purrr::pluck(&quot;ta1&quot;, &quot;top_50&quot;) %&gt;%
  ComplexHeatmap::draw(heatmap_legend_side = &quot;left&quot;, annotation_legend_side = &quot;bottom&quot;)</code></pre>
<p><img src="figure/100_global_description.Rmd/hierarchical_clustering_new_vars-1.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-hierarchical_clustering_new_vars-1">
Past versions of hierarchical_clustering_new_vars-1.png
</button>
</p>
<div id="fig-hierarchical_clustering_new_vars-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/hierarchical_clustering_new_vars-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>In this heatmap with hierarchical clustering, we can clearly see that
the three bacterial groups correspond to the defined clusters. Moreover,
we can clearly see the association between the variables vaginotype and
vaginotype_condition and species composition.</p>
</div>
<div id="section-7" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
</div>
<div id="alpha-diversity---nmds" class="section level2">
<h2>Alpha Diversity - NMDS</h2>
<p>NMDS (Non-metric Multidimensional Scaling) analysis is a commonly
used non-metric ordination technique that helps visualize the similarity
or dissimilarity between samples based on a distance or similarity
matrix. In the context of microbiome data, this technique is
particularly useful for identifying patterns and significant changes in
microbiome composition among different samples or groups.</p>
<pre class="r"><code>mre &lt;- metar::virgo_nmds(mre, top_n = 50, save_files = FALSE)
metar::get_taxa(mre, &quot;virgo&quot;, &quot;nmds&quot;) %&gt;%
  purrr::pluck(&quot;ta1&quot;, &quot;top_50&quot;, &quot;all_adonis&quot;) %&gt;%
  dplyr::arrange(Pr..F.)</code></pre>
<pre><code>#&gt; # A tibble: 10 × 6
#&gt;    id                      Df SumOfSqs      R2       F Pr..F.
#&gt;    &lt;chr&gt;                &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt;
#&gt;  1 Treatment_group          6    4.65  0.0268    2.18   0.001
#&gt;  2 Treatment_group_1        4    4.17  0.0240    2.93   0.001
#&gt;  3 pass_reads_quantile      3   33.9   0.195    38.6    0.001
#&gt;  4 vaginotype               2   78.3   0.452   197.     0.001
#&gt;  5 vaginotype_condition     1   46.3   0.267   175.     0.001
#&gt;  6 cluster                  2   78.3   0.452   197.     0.001
#&gt;  7 Heme_test                1    1.16  0.00668   3.22   0.005
#&gt;  8 Medical_center           1    0.718 0.00414   1.99   0.065
#&gt;  9 Arm                      1    0.524 0.00302   1.45   0.186
#&gt; 10 Timepoint                5    1.23  0.00711   0.681  0.916</code></pre>
<p>The Adonis test indicates that there are significant differences in
the taxonomic composition between the groups of the categorical
variables: <strong>Treatment_group_1</strong>,
<strong>pass_reads_quantile</strong>, <strong>vaginotype</strong>, and
<strong>vaginotype_condition</strong>.</p>
</div>
<div id="section-8" class="section level2 tabset">
<h2 class="tabset"></h2>
<pre class="r"><code>metar::get_cat(mre) %&gt;%
  dplyr::pull(1) %&gt;%
  .[-1] %&gt;% # Remove &quot;Treatment_group&quot;
  purrr::walk( ~ {
    cat(&#39;\n\n### `&#39;, .x, &#39;`\n\n&#39;)
    metar::get_taxa(mre, &quot;virgo&quot;, &quot;nmds&quot;) %&gt;%
      purrr::pluck(&quot;ta1&quot;, &quot;top_50&quot;, &quot;categorical&quot;, .x) %&gt;%
      plot()
  })</code></pre>
<div id="arm-5" class="section level3">
<h3><code>Arm</code></h3>
<p><img src="figure/100_global_description.Rmd/nmds_groups-1.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-nmds_groups-1">
Past versions of nmds_groups-1.png
</button>
</p>
<div id="fig-nmds_groups-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/nmds_groups-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="heme_test-5" class="section level3">
<h3><code>Heme_test</code></h3>
<p><img src="figure/100_global_description.Rmd/nmds_groups-2.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-nmds_groups-2">
Past versions of nmds_groups-2.png
</button>
</p>
<div id="fig-nmds_groups-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/nmds_groups-2.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="medical_center-5" class="section level3">
<h3><code>Medical_center</code></h3>
<p><img src="figure/100_global_description.Rmd/nmds_groups-3.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-nmds_groups-3">
Past versions of nmds_groups-3.png
</button>
</p>
<div id="fig-nmds_groups-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/nmds_groups-3.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="treatment_group_1-5" class="section level3">
<h3><code>Treatment_group_1</code></h3>
<p><img src="figure/100_global_description.Rmd/nmds_groups-4.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-nmds_groups-4">
Past versions of nmds_groups-4.png
</button>
</p>
<div id="fig-nmds_groups-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/nmds_groups-4.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pass_reads_quantile-5" class="section level3">
<h3><code>pass_reads_quantile</code></h3>
<p><img src="figure/100_global_description.Rmd/nmds_groups-5.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-nmds_groups-5">
Past versions of nmds_groups-5.png
</button>
</p>
<div id="fig-nmds_groups-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/nmds_groups-5.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="vaginotype-2" class="section level3">
<h3><code>vaginotype</code></h3>
<p><img src="figure/100_global_description.Rmd/nmds_groups-6.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-nmds_groups-6">
Past versions of nmds_groups-6.png
</button>
</p>
<div id="fig-nmds_groups-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/nmds_groups-6.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="vaginotype_condition-2" class="section level3">
<h3><code>vaginotype_condition</code></h3>
<p><img src="figure/100_global_description.Rmd/nmds_groups-7.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-nmds_groups-7">
Past versions of nmds_groups-7.png
</button>
</p>
<div id="fig-nmds_groups-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/nmds_groups-7.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="section-9" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
<p>The nmds plots also indicate that there are significant differences
between the levels of the categorical variables:
<strong>pass_reads_quantile</strong>, <strong>vaginotype</strong>, and
<strong>vaginotype_condition</strong>. Moreover, the species
contributing to cluster differentiation can be seen. While <strong>G.
Vaginalis</strong> and <strong>A. Vaginae</strong> contribute to the
formation of cluster 2 (and to the appearance of vaginosis), <strong>L.
Crispatus</strong> and <strong>L. Iners</strong> contribute to their
respective clusters.</p>
</div>
<div
id="longitudinal-stability-of-vaginotype-within-individual-patients"
class="section level2 tabset">
<h2 class="tabset">Longitudinal Stability of Vaginotype within
Individual Patients</h2>
<p>At this point, it is clear that the categorical variable “vaginotype”
is a significant component of the observed compositional variability. To
further investigate this phenotypic characteristic, we ask the question
of whether the vaginotype remains stable longitudinally within the same
patient.</p>
<p>First, we generated alluvial plots for each categorical variable.</p>
<pre class="r"><code>library(ggalluvial)
metar::get_cat(mre) %&gt;%
  dplyr::pull(1) %&gt;%
  .[-1] %&gt;% # Remove &quot;Treatment_group&quot;
  purrr::walk(~ {
    cat(&#39;\n\n### `&#39;, .x, &#39;`\n\n&#39;)
    p &lt;- 
      metar::get_meta(mre) %&gt;%
      dplyr::select(PatientID, !!dplyr::sym(.x), Timepoint) %&gt;%
      ggplot(
        aes(
          as.factor(Timepoint),
          stratum = !!dplyr::sym(.x),
          alluvium = PatientID,
          fill = !!dplyr::sym(.x),
          label = !!dplyr::sym(.x)
        )
      ) +
      scale_x_discrete(expand = c(.1, 0)) +
      geom_flow(width = 1 / 4) +
      geom_stratum(alpha = .5, width = 1 / 4) +
      geom_text(stat = &quot;stratum&quot;, size = 4) +
      theme_minimal() +
      theme(legend.position = &quot;none&quot;)
    
    plot(p)
  })</code></pre>
<div id="arm-6" class="section level3">
<h3><code>Arm</code></h3>
<p><img src="figure/100_global_description.Rmd/alluvial_plot-1.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alluvial_plot-1">
Past versions of alluvial_plot-1.png
</button>
</p>
<div id="fig-alluvial_plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alluvial_plot-1.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="heme_test-6" class="section level3">
<h3><code>Heme_test</code></h3>
<p><img src="figure/100_global_description.Rmd/alluvial_plot-2.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alluvial_plot-2">
Past versions of alluvial_plot-2.png
</button>
</p>
<div id="fig-alluvial_plot-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alluvial_plot-2.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="medical_center-6" class="section level3">
<h3><code>Medical_center</code></h3>
<p><img src="figure/100_global_description.Rmd/alluvial_plot-3.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alluvial_plot-3">
Past versions of alluvial_plot-3.png
</button>
</p>
<div id="fig-alluvial_plot-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alluvial_plot-3.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="treatment_group_1-6" class="section level3">
<h3><code>Treatment_group_1</code></h3>
<p><img src="figure/100_global_description.Rmd/alluvial_plot-4.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alluvial_plot-4">
Past versions of alluvial_plot-4.png
</button>
</p>
<div id="fig-alluvial_plot-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alluvial_plot-4.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pass_reads_quantile-6" class="section level3">
<h3><code>pass_reads_quantile</code></h3>
<p><img src="figure/100_global_description.Rmd/alluvial_plot-5.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alluvial_plot-5">
Past versions of alluvial_plot-5.png
</button>
</p>
<div id="fig-alluvial_plot-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alluvial_plot-5.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="vaginotype-3" class="section level3">
<h3><code>vaginotype</code></h3>
<p><img src="figure/100_global_description.Rmd/alluvial_plot-6.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alluvial_plot-6">
Past versions of alluvial_plot-6.png
</button>
</p>
<div id="fig-alluvial_plot-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alluvial_plot-6.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="vaginotype_condition-3" class="section level3">
<h3><code>vaginotype_condition</code></h3>
<p><img src="figure/100_global_description.Rmd/alluvial_plot-7.png" width="1260" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alluvial_plot-7">
Past versions of alluvial_plot-7.png
</button>
</p>
<div id="fig-alluvial_plot-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/OriolGEM/OriolGEM.github.io/blob/e36d2e8b7f46d25c1a7f20e09311bf664f4787c7/docs/figure/100_global_description.Rmd/alluvial_plot-7.png" target="_blank">e36d2e8</a>
</td>
<td>
OriolGEM
</td>
<td>
2024-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="section-10" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
<p>We can observe that the vaginotype variable does not remain stable
over time. Some patients start with a potential vaginosis phenotype and
then transition to a healthy phenotype, while others transition from a
healthy phenotype to vaginosis. Additionally, some patients go through
both stages cyclically.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>#&gt; R version 4.1.2 (2021-11-01)
#&gt; Platform: x86_64-pc-linux-gnu (64-bit)
#&gt; Running under: Ubuntu 22.04.4 LTS
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so
#&gt; 
#&gt; locale:
#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#&gt;  [3] LC_TIME=es_ES.UTF-8        LC_COLLATE=en_US.UTF-8    
#&gt;  [5] LC_MONETARY=es_ES.UTF-8    LC_MESSAGES=en_US.UTF-8   
#&gt;  [7] LC_PAPER=es_ES.UTF-8       LC_NAME=C                 
#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
#&gt; [11] LC_MEASUREMENT=es_ES.UTF-8 LC_IDENTIFICATION=C       
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     
#&gt; 
#&gt; other attached packages:
#&gt; [1] ggalluvial_0.12.5 ggplot2_3.5.1     magrittr_2.0.3   
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;   [1] utf8_1.2.4             tidyselect_1.2.1       htmlwidgets_1.6.4     
#&gt;   [4] grid_4.1.2             gmp_0.7-4              aws.signature_0.6.0   
#&gt;   [7] munsell_0.5.1          codetools_0.2-20       effectsize_0.8.8      
#&gt;  [10] withr_3.0.0            colorspace_2.1-0       Biobase_2.54.0        
#&gt;  [13] phyloseq_1.38.0        logger_0.3.0           highr_0.10            
#&gt;  [16] knitr_1.46             rstudioapi_0.16.0      stats4_4.1.2          
#&gt;  [19] ggsignif_0.6.4         ggside_0.3.1           labeling_0.4.3        
#&gt;  [22] git2r_0.33.0           GenomeInfoDbData_1.2.7 polyclip_1.10-6       
#&gt;  [25] bit64_4.0.5            farver_2.1.2           datawizard_0.10.0     
#&gt;  [28] rhdf5_2.38.1           rprojroot_2.0.4        vctrs_0.6.5           
#&gt;  [31] generics_0.1.3         xfun_0.44              timechange_0.3.0      
#&gt;  [34] BWStest_0.2.3          R6_2.5.1               doParallel_1.0.17     
#&gt;  [37] GenomeInfoDb_1.30.1    clue_0.3-65            bitops_1.0-7          
#&gt;  [40] rhdf5filters_1.6.0     cachem_1.1.0           promises_1.3.0        
#&gt;  [43] scales_1.3.0           vroom_1.6.5            nnet_7.3-19           
#&gt;  [46] gtable_0.3.5           multcompView_0.1-10    workflowr_1.7.1       
#&gt;  [49] rlang_1.1.3            zeallot_0.1.0          GlobalOptions_0.1.2   
#&gt;  [52] PMCMRplus_1.9.10       splines_4.1.2          rstatix_0.7.2         
#&gt;  [55] lazyeval_0.2.2         broom_1.0.6            prismatic_1.1.2       
#&gt;  [58] checkmate_2.3.1        yaml_2.3.8             reshape2_1.4.4        
#&gt;  [61] abind_1.4-5            backports_1.5.0        httpuv_1.6.15         
#&gt;  [64] Hmisc_5.1-2            ggmosaic_0.3.3         tools_4.1.2           
#&gt;  [67] gplots_3.1.3.1         jquerylib_0.1.4        biomformat_1.22.0     
#&gt;  [70] RColorBrewer_1.1-3     BiocGenerics_0.40.0    Rcpp_1.0.12           
#&gt;  [73] plyr_1.8.9             base64enc_0.1-3        zlibbioc_1.40.0       
#&gt;  [76] purrr_1.0.2            RCurl_1.98-1.14        ggpubr_0.6.0          
#&gt;  [79] rpart_4.1.23           GetoptLong_1.0.5       correlation_0.8.4     
#&gt;  [82] S4Vectors_0.32.4       ggrepel_0.9.5          cluster_2.1.6         
#&gt;  [85] fs_1.6.4               here_1.0.1             magick_2.8.3          
#&gt;  [88] data.table_1.15.4      circlize_0.4.16        mvtnorm_1.2-5         
#&gt;  [91] whisker_0.4.1          matrixStats_1.3.0      hms_1.1.3             
#&gt;  [94] patchwork_1.2.0        evaluate_0.23          IRanges_2.28.0        
#&gt;  [97] gridExtra_2.3          shape_1.4.6.1          compiler_4.1.2        
#&gt; [100] tibble_3.2.1           KernSmooth_2.23-24     ggstatsplot_0.12.3    
#&gt; [103] crayon_1.5.2           htmltools_0.5.8.1      mgcv_1.9-1            
#&gt; [106] later_1.3.2            tzdb_0.4.0             Formula_1.2-5         
#&gt; [109] tidyr_1.3.1            lubridate_1.9.3        aws.s3_0.3.21         
#&gt; [112] SuppDists_1.1-9.7      kSamples_1.2-10        tweenr_2.0.3          
#&gt; [115] ComplexHeatmap_2.18.0  MASS_7.3-60.0.1        boot_1.3-30           
#&gt; [118] readr_2.1.5            Matrix_1.3-4           ade4_1.7-22           
#&gt; [121] car_3.1-2              wesanderson_0.3.7      permute_0.9-7         
#&gt; [124] cli_3.6.2              parallel_4.1.2         insight_0.19.11       
#&gt; [127] igraph_2.0.3           forcats_1.0.0          pkgconfig_2.0.3       
#&gt; [130] metar_0.1.5            statsExpressions_1.5.4 foreign_0.8-86        
#&gt; [133] plotly_4.10.4          xml2_1.3.6             paletteer_1.6.0       
#&gt; [136] foreach_1.5.2          bslib_0.7.0            multtest_2.50.0       
#&gt; [139] XVector_0.34.0         snakecase_0.11.1       stringr_1.5.1         
#&gt; [142] digest_0.6.35          parameters_0.21.7      janitor_2.2.0         
#&gt; [145] vegan_2.6-6.1          Biostrings_2.62.0      rmarkdown_2.27        
#&gt; [148] htmlTable_2.4.2        curl_5.2.1             gtools_3.9.5          
#&gt; [151] rjson_0.2.21           lifecycle_1.0.4        nlme_3.1-164          
#&gt; [154] jsonlite_1.8.8         Rhdf5lib_1.16.0        carData_3.0-5         
#&gt; [157] viridisLite_0.4.2      fansi_1.0.6            pillar_1.9.0          
#&gt; [160] lattice_0.22-6         fastmap_1.2.0          httr_1.4.7            
#&gt; [163] survival_3.6-4         glue_1.7.0             bayestestR_0.13.2     
#&gt; [166] png_0.1-8              iterators_1.0.14       bit_4.0.5             
#&gt; [169] ggforce_0.4.2          stringi_1.8.4          sass_0.4.9            
#&gt; [172] performance_0.11.0     rematch2_2.1.2         caTools_1.18.2        
#&gt; [175] memoise_2.0.1          Rmpfr_0.9-5            dplyr_1.1.4           
#&gt; [178] ape_5.8</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
