<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Oriol Careta Borràs" />

<meta name="date" content="2024-04-10" />

<title>CHIR HIV RV306 Vaginal - AIDSVax Effect</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">2024_cihr_hiv_rv306</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="004_throughput_analysis.html">Throughput Analysis</a>
</li>
<li>
  <a href="005_mre_initialization.html">Init mre object</a>
</li>
<li>
  <a href="007_phylosmith_abundance.html">Phylosmith abundance exploration</a>
</li>
<li>
  <a href="008_phylosmith_diversity.html">Phylosmith diversity exploration</a>
</li>
<li>
  <a href="009_quality_control.html">Quality_control</a>
</li>
<li>
  <a href="100_global_description.html">Global description</a>
</li>
<li>
  <a href="101_baseline_timepoints.html">Baseline timepoints</a>
</li>
<li>
  <a href="102_baseline_description.html">Baseline description</a>
</li>
<li>
  <a href="103_AIDSVax_effect.html">AIDXVax effect</a>
</li>
<li>
  <a href="104_groups_comparison.html">Groups Comparison</a>
</li>
<li>
  <a href="105_longitudinal_sudy_t1.html">Longitudinal Study of Group 1</a>
</li>
<li>
  <a href="105_longitudinal_sudy_t2.html">Longitudinal Study of Group 2</a>
</li>
<li>
  <a href="105_longitudinal_sudy_t3.html">Longitudinal Study of Group 3</a>
</li>
<li>
  <a href="105_longitudinal_sudy_t4A.html">Longitudinal Study of Group 4A</a>
</li>
<li>
  <a href="105_longitudinal_sudy_t4B.html">Longitudinal Study of Group 4B</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">CHIR HIV RV306 Vaginal - AIDSVax
Effect</h1>
<h4 class="author">Oriol Careta Borràs</h4>
<h4 class="date">2024-04-10</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-04-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong>
<code>2024_chir_hiv_rv306_vaginal/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240321code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240321)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240321code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240321)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
adonis_table
</li>
<li>
gene_richness
</li>
<li>
hierarchical_clustering
</li>
<li>
load_data
</li>
<li>
lon_gene_richness
</li>
<li>
nmds_groups
</li>
<li>
PCoA
</li>
<li>
pred_taxa
</li>
<li>
rarefraction_plots
</li>
<li>
run_ancom
</li>
<li>
run_lefse
</li>
<li>
session-info-chunk-inserted-by-workflowr
</li>
<li>
top50_abundance_barplot
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory
<code>103_AIDSVax_effect_cache</code> and re-run the analysis. To have
workflowr automatically delete the cache directory prior to building the
file, set <code>delete_cache = TRUE</code> when running
<code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong412fedd">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 412fedd
</a>
</p>
</div>
<div id="strongRepositoryversionstrong412fedd"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 412fedd.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/009_quality_control_cache/
    Ignored:    analysis/100_global_description_cache/
    Ignored:    analysis/101_baseline_timepoints_cache/
    Ignored:    analysis/102_baseline_description_cache/
    Ignored:    analysis/103_AIDSVax_effect_cache/
    Ignored:    analysis/104_groups_comparison_cache/
    Ignored:    analysis/105_longitudinal_study_cache/
    Ignored:    analysis/105_longitudinal_sudy_t1_cache/
    Ignored:    analysis/105_longitudinal_sudy_t2_cache/
    Ignored:    analysis/105_longitudinal_sudy_t3_cache/
    Ignored:    analysis/105_longitudinal_sudy_t4A_cache/
    Ignored:    analysis/105_longitudinal_sudy_t4B_cache/
    Ignored:    analysis/Study_1.png
    Ignored:    analysis/Study_2.png
    Ignored:    analysis/Study_3.png
    Ignored:    analysis/Study_4A.png
    Ignored:    analysis/Study_4B.png
    Ignored:    code/.RData
    Ignored:    code/.Rhistory

Untracked files:
    Untracked:  analysis/004_throughput_analysis.Rmd
    Untracked:  analysis/005_mre_initialization.Rmd
    Untracked:  analysis/007_phylosmith_abundance.Rmd
    Untracked:  analysis/008_phylosmith_diversity.Rmd
    Untracked:  analysis/009_quality_control.Rmd
    Untracked:  analysis/100_global_description.Rmd
    Untracked:  analysis/101_baseline_timepoints.Rmd
    Untracked:  analysis/102_baseline_description.Rmd
    Untracked:  analysis/103_AIDSVax_effect.Rmd
    Untracked:  analysis/104_groups_comparison.Rmd
    Untracked:  analysis/105_longitudinal_sudy_t1.Rmd
    Untracked:  analysis/105_longitudinal_sudy_t2.Rmd
    Untracked:  analysis/105_longitudinal_sudy_t3.Rmd
    Untracked:  analysis/105_longitudinal_sudy_t4A.Rmd
    Untracked:  analysis/105_longitudinal_sudy_t4B.Rmd
    Untracked:  analysis/Study.jpg
    Untracked:  code/000_s3_bucket_initialization.Rmd
    Untracked:  code/001_metadata_check.Rmd
    Untracked:  code/002_datatransfer.Rmd
    Untracked:  code/003_metadata_vars_selection.Rmd
    Untracked:  code/006_populate_mre.Rmd
    Untracked:  data/RV306_CVM_2022_11_08_lims.xlsx
    Untracked:  data/RV306_CVM_ShippingManifest_2022_SB_clinical.xlsx
    Untracked:  data/preprocessed/

Unstaged changes:
    Modified:   analysis/_site.yml
    Modified:   analysis/about.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>We are interested in comparing samples of time-points 0, and 14 in
terms of microbiome composition and function to answer the question how
does the application of AIDSVax affect microbiome composition?.</p>
<div id="mre-initialization-and-filter-baseline-samples"
class="section level2">
<h2>MRE initialization and filter baseline samples</h2>
<pre class="r"><code>mre &lt;- readr::read_rds(here::here(&quot;data&quot;, &quot;preprocessed&quot;, &quot;mre_vaginotype.rds&quot;))

## mre init &amp; filter
mre &lt;- 
  metar::get_meta(mre) %&gt;%
  dplyr::filter(Timepoint %in% c(0, 14)) %&gt;%
  dplyr::pull(SampleID) %&gt;%
  metar::filter_samples(mre, sample_ids = .)

mre</code></pre>
<pre><code>#&gt; MetarSet-class experimental-level object 
#&gt;   [MetadataSet     ]: 231 samples and 9 variables 
#&gt;   [DiversitySet    ]: 
#&gt;      ✖ igc: 0 observations and 0 samples 
#&gt;      ✔ virgo: 7109 observations and 231 samples 
#&gt;      ✖ dada2: 0 observations and 0 samples 
#&gt;   [TaxaSet         ]: 
#&gt;      ✖ metaphlan: no phyloseq object available 
#&gt;      ✖ kraken: no phyloseq object available 
#&gt;      ✖ bracken: no phyloseq object available 
#&gt;      ✖ motus: no phyloseq object available 
#&gt;      ✔ virgo: phyloseq object with 255 taxa and 231 samples 
#&gt;      ✖ dada2: no phyloseq object available 
#&gt;   [GeneFunctionSet ]: 
#&gt;      ✖ humann: no functional data aviable 
#&gt;      ✖ igc: no functional data aviable 
#&gt;      ✔ virgo: 4333 kegg_id, 0 cog_id, 0 eggnog_id, 0 ec_id and 0 metacyc_id 
#&gt;      ✖ fmap: no functional data aviable 
#&gt;      ✖ picrust2: no functional data aviable 
#&gt;      ✖ tax4fun: no functional data aviable 
#&gt;      ✖ phylolisft: no functional data aviable</code></pre>
<p>There are 231 samples from 118 patients which were collected at
time-points 0 and 14.</p>
</div>
<div id="gene-richness---rarefaction" class="section level2 tabset">
<h2 class="tabset">Gene richness - Rarefaction</h2>
<p>Note that diversity analysis here are delicate to perform due to the
limitations in sampling.</p>
<p>From the taxonomic composition we’ll derive some alpha-diversity
ecological indices and compare them among groups.</p>
<p>We also want to use the VIRGO catalog mapping-derived gene richness.
However, this can be problematic for gene richness since the initial
reads per sample is very low in some cases. This low numbers may be
enough to describe the taxonomic composition, specially if the
microbiome is “simple”, but may fail to capture richness/diversity.</p>
<pre class="r"><code>## run rarefaction
mre &lt;- metar::virgo_rarefaction(mre, save_files = FALSE)

## plot rarefaction
metar::get_cat(mre) %&gt;%
  dplyr::pull(1) %&gt;%
  purrr::walk(~ {
    cat(&#39;\n\n### `&#39;, .x, &#39;`\n\n&#39;)
    metar::get_diversity(mre, type = &quot;virgo&quot;, res_slot = &quot;rarefaction&quot;) %&gt;%
      purrr::pluck(&quot;categorical&quot;, .x, &quot;rarefaction&quot;) %&gt;%
      plot()
  })</code></pre>
<div id="treatment_group" class="section level3">
<h3><code>Treatment_group</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/rarefraction_plots-1.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="arm" class="section level3">
<h3><code>Arm</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/rarefraction_plots-2.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="heme_test" class="section level3">
<h3><code>Heme_test</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/rarefraction_plots-3.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="medical_center" class="section level3">
<h3><code>Medical_center</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/rarefraction_plots-4.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="pass_reads_quantile" class="section level3">
<h3><code>pass_reads_quantile</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/rarefraction_plots-5.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="vaginotype" class="section level3">
<h3><code>vaginotype</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/rarefraction_plots-6.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="vaginotype_condition" class="section level3">
<h3><code>vaginotype_condition</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/rarefraction_plots-7.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="section" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
<p>The rarefraction curves (used to present relationship between gene
richness and sequencing depth) show no apparent difference between
groups in terms of diversity. Of note, rarefaction curve shows how far
we are from asymptotically saturating richness signal from this catalog
at the sampling depth we have (after filtering human sequences). Thus,
we need to be aware of the limitations of gene richness analysis and
interpret with caution. Of note, samples enriched in <strong>G.
Vaginalis</strong> (and thus, with a vaginosis profile) seem to have a
higher sample depth and gene richness compared to the other ones.</p>
</div>
<div id="gene-richness---boxplots" class="section level2 tabset">
<h2 class="tabset">Gene richness - Boxplots</h2>
<pre class="r"><code>## plot gene richness
metar::get_cat(mre) %&gt;%
  dplyr::pull(1) %&gt;%
  purrr::walk(~ {
    cat(&#39;\n\n### `&#39;, .x, &#39;`\n\n&#39;)
    metar::get_diversity(mre, type = &quot;virgo&quot;, res_slot = &quot;rarefaction&quot;) %&gt;%
      purrr::pluck(&quot;categorical&quot;, .x, &quot;boxplots&quot;, &quot;GeneNumber&quot;) %&gt;%
      plot()
  })</code></pre>
<div id="treatment_group-1" class="section level3">
<h3><code>Treatment_group</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/gene_richness-1.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="arm-1" class="section level3">
<h3><code>Arm</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/gene_richness-2.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="heme_test-1" class="section level3">
<h3><code>Heme_test</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/gene_richness-3.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="medical_center-1" class="section level3">
<h3><code>Medical_center</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/gene_richness-4.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="pass_reads_quantile-1" class="section level3">
<h3><code>pass_reads_quantile</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/gene_richness-5.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="vaginotype-1" class="section level3">
<h3><code>vaginotype</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/gene_richness-6.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="vaginotype_condition-1" class="section level3">
<h3><code>vaginotype_condition</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/gene_richness-7.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="section-1" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
<p>The gene richness boxplots indicate that there are no significant
differences between groups in terms of gene richness, except when
separating by the number of QPass reads by quantiles, where the gene
richness on the 4th quantile group is significantly higher. This could
be explained due to the fact that the sample depth is higher on this
group which favors gene richness. There are also significant differences
between vaginotype groups. Ther also differences in gene richness when
comparing vaginotype groups.</p>
</div>
<div id="alpha-diversity---longitudinal" class="section level2 tabset">
<h2 class="tabset">Alpha Diversity - Longitudinal</h2>
<p>To further explore the aforementioned observations, we will verify
whether gene diversity varies across timepoints for the categorical
variables.</p>
<pre class="r"><code>metar::get_cat(mre) %&gt;%
  dplyr::pull(1) %&gt;%
  purrr::walk(~ {
    cat(&#39;\n\n### `&#39;, .x, &#39;`\n\n&#39;)
    metar::get_diversity(mre, type = &quot;virgo&quot;, res_slot = &quot;rarefaction&quot;) %&gt;%
      purrr::pluck(&quot;longitudinal_rel&quot;, &quot;categorical&quot;, .x) %&gt;%
      plot()
  })</code></pre>
<div id="treatment_group-2" class="section level3">
<h3><code>Treatment_group</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/lon_gene_richness-1.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="arm-2" class="section level3">
<h3><code>Arm</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/lon_gene_richness-2.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="heme_test-2" class="section level3">
<h3><code>Heme_test</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/lon_gene_richness-3.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="medical_center-2" class="section level3">
<h3><code>Medical_center</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/lon_gene_richness-4.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="pass_reads_quantile-2" class="section level3">
<h3><code>pass_reads_quantile</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/lon_gene_richness-5.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="vaginotype-2" class="section level3">
<h3><code>vaginotype</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/lon_gene_richness-6.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="vaginotype_condition-2" class="section level3">
<h3><code>vaginotype_condition</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/lon_gene_richness-7.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="section-2" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
<p>In these graphs, we can see the changes in the relative gene richness
over time of samples separated by categorical variable groups. It seems
that microbiome stability is higher in the placebo arm compared to the
active arm.</p>
</div>
<div id="pcoa-based-on-bray-curtis" class="section level2">
<h2>PCoA based on Bray Curtis</h2>
<pre class="r"><code>ps &lt;- metar::get_phyloseq(mre, type = &quot;virgo&quot;)
ps_RA &lt;- phyloseq::transform_sample_counts(ps, function(x) x / sum(x))
bray_pcoa  &lt;-
  phyloseq::ordinate(physeq = ps_RA, method = &quot;PCoA&quot;, distance = &quot;bray&quot;)

create_pcoa_plot &lt;- function(ordination_obj, title, a, b) {
  plot &lt;- phyloseq::plot_ordination(
    physeq = ps_RA,
    ordination = ordination_obj,
    color = &quot;vaginotype&quot;,
    shape = &quot;Arm&quot;,
    axes = c(a,b)
  ) +
  geom_point(size = 2.5) +
  scale_color_manual(values = c(&quot;deeppink4&quot;, &quot;coral&quot;, &quot;cadetblue4&quot;)) +
  scale_shape_manual(values = c(16, 2)) +
  theme_light() +
  ggtitle(title)
  
  return(plot)
}

pcoa_plot_PCo1_PCo2 &lt;- create_pcoa_plot(bray_pcoa, &quot;PCo1 vs PCo2&quot;, 1, 2)
pcoa_plot_PCo1_PCo3 &lt;- create_pcoa_plot(bray_pcoa, &quot;PCo1 vs PCo3&quot;, 1, 3)
pcoa_plot_PCo2_PCo3 &lt;- create_pcoa_plot(bray_pcoa, &quot;PCo2 vs PCo3&quot;, 2, 3)

gridExtra::grid.arrange(pcoa_plot_PCo1_PCo2, pcoa_plot_PCo1_PCo3, pcoa_plot_PCo2_PCo3, nrow = 1)</code></pre>
<p><img src="figure/103_AIDSVax_effect.Rmd/PCoA-1.png" width="3600" style="display: block; margin: auto;" /></p>
<p>In the PCoA plots, it can be seen that the compositional variables
that affect each cluster. PCo 1 and 2 explain the 84.7% of the
variability (48.6% and 36.1% respectively), while PCo3 only explains the
5.4% of the variability. From the plots we can see that PCo1 is the one
that explains the separation of cluster 1 (L. Crispatus) and cluster 3
(L. Iners) from cluster 2 (G. Vaginalis), while PCo2 explains the
separation of cluster 2 and 3 from cluster 1. On the other hand, PCo3 is
not useful to discriminate between clusters. It can also be seen
that</p>
</div>
<div id="beta-diversity---species-composition" class="section level2">
<h2>Beta-Diversity - Species Composition</h2>
<p>Let’s visualize composition at the species level using barplots.</p>
<pre class="r"><code>## run barplots 
mre &lt;- metar::virgo_barplots(mre, top_n = 50, save_files = FALSE)

## Plot virgo barplots
mre %&gt;% 
  metar::get_taxa(&quot;virgo&quot;, &quot;barplots&quot;) %&gt;%
  purrr::pluck(&quot;ta1&quot;, &quot;top_50&quot;, &quot;rel_abundance&quot;, &quot;bray_NMDS1_order_barplot&quot;) +
  theme(axis.text.x = element_blank())</code></pre>
<p><img src="figure/103_AIDSVax_effect.Rmd/top50_abundance_barplot-1.png" width="3600" style="display: block; margin: auto;" /></p>
<p>The clustering barplots depicting taxonomic diversity per sample
reveal the presence of three distinct groups:</p>
<ul>
<li>The first group is primarily characterized by the dominance of
<strong>Gardnerella vaginalis</strong>, occasionally accompanied by
co-dominance with <strong>Atopobium vaginae</strong>. Notably, this
group exhibits a higher level of taxonomic diversity compared to the
other two groups. Both <strong>Gardnerella vaginalis</strong>, a
gram-negative facultative anaerobic bacterium, and <strong>Atopobium
vaginae</strong>, a gram-positive strict anaerobic bacterium, have been
associated with bacterial vaginosis (BV) and its characteristic
symptoms.</li>
<li>The second group, is characterized by a strong dominance of
<strong>Lactobacillus iners</strong>. <strong>Lactobacillus
iners</strong> is a common bacterium found in the vaginal microbiota,
playing a vital role in maintaining vaginal health. It helps to maintain
an acidic pH, creating an unfavorable environment for harmful
microorganisms. <strong>Lactobacillus iners</strong> contributes to the
stability and balance of the vaginal microbiota, promoting overall
vaginal health and potentially protecting against infections like
bacterial vaginosis.</li>
<li>The third group is mainly characterized by a dominant presence of
<strong>Lactobacillus crispatus</strong>, although some samples in this
group also show high levels of <strong>Lactobacillus iners</strong>.
<strong>Lactobacillus crispatus</strong> is a beneficial bacterium
commonly found in the vaginal microbiota. It helps maintain a healthy
vaginal environment by producing lactic acid and supporting acidity.
Some samples also have some codominance of <strong>Lactobacillus
jensenii</strong>. <strong>Lactobacillus jensenii</strong> is a
gram-positive bacterium commonly found in the vaginal microbiota,
particularly in women of reproductive age. It contributes to maintaining
vaginal health by producing lactic acid and creating an acidic
environment, which helps inhibit the growth of harmful bacteria.
Lactobacillus jensenii is known for its ability to adhere to vaginal
epithelial cells, which may enhance its protective effects against
infections and maintain the balance of the vaginal microbiota.</li>
</ul>
<pre class="r"><code>## Count of two predominant taxas per sample
mre %&gt;% 
  metar::get_taxa(&quot;virgo&quot;, &quot;barplots&quot;) %&gt;%
  purrr::pluck(&quot;ta1&quot;, &quot;top_50&quot;, &quot;rel_abundance&quot;, &quot;bray_NMDS1_order_barplot&quot;, &quot;data&quot;) %&gt;%
  dplyr::group_by(SampleID) %&gt;%
  dplyr::slice_max(Abundance, n = 2) %&gt;%
  dplyr::ungroup() %&gt;%
  dplyr::count(Species, sort = TRUE)</code></pre>
<pre><code>#&gt; # A tibble: 20 × 2
#&gt;    Species                           n
#&gt;    &lt;fct&gt;                         &lt;int&gt;
#&gt;  1 Lactobacillus_iners             119
#&gt;  2 Gardnerella_vaginalis           108
#&gt;  3 Lactobacillus_crispatus         100
#&gt;  4 Atopobium_vaginae                52
#&gt;  5 Lactobacillus_jensenii           36
#&gt;  6 Lactobacillus_gasseri             8
#&gt;  7 BVAB1                             7
#&gt;  8 Prevotella_amnii                  7
#&gt;  9 Prevotella_bivia                  5
#&gt; 10 Streptococcus_anginosus           4
#&gt; 11 Streptococcus_agalactiae          3
#&gt; 12 Lactobacillus_johnsonii           2
#&gt; 13 Streptococcus_urinalis            2
#&gt; 14 Prevotella_timonensis             2
#&gt; 15 Mobiluncus_mulieris               2
#&gt; 16 Ureaplasma_urealyticum            1
#&gt; 17 Staphylococcus_epidermidis        1
#&gt; 18 Corynebacterium_aurimucosum       1
#&gt; 19 Prevotella_buccalis               1
#&gt; 20 Lactobacillus_kefiranofaciens     1</code></pre>
<p>In this table, we can see the to most dominant species of each sample
summed up. It can be seen that the most abundant species are
<strong>Gardnerella vaginalis</strong>, <strong>Lactobacillus
iners</strong> and <strong>Lactobacillus crispatus</strong>, followed by
<strong>Atopobium vaginae</strong>, which on the abundance barplot was
found in codominance with <strong>Gardnerella vaginalis</strong> on some
samples.</p>
</div>
<div id="beta-diversity---cluestering" class="section level2">
<h2>Beta-Diversity - Cluestering</h2>
<p>In the previous barplots, the x-axis is defined according to ward.D2,
bray-based clustering. We’ll now transform abundances and project them
into a heatmap, to be able to visualize association with variables of
interes.</p>
<pre class="r"><code>## run heatmap
mre &lt;- metar::virgo_heatmap(mre, top_n = 50, save_files = FALSE)

## plot heatmap
library(ComplexHeatmap)
mre %&gt;%
  metar::get_taxa(&quot;virgo&quot;, &quot;heatmaps&quot;) %&gt;%
  purrr::pluck(&quot;ta1&quot;, &quot;top_50&quot;) %&gt;%
  ComplexHeatmap::draw(heatmap_legend_side = &quot;left&quot;, annotation_legend_side = &quot;bottom&quot;)</code></pre>
<p><img src="figure/103_AIDSVax_effect.Rmd/hierarchical_clustering-1.png" width="1260" style="display: block; margin: auto;" /></p>
<p>In this hierarchical clustering, we can clearly see the three groups,
but no clear associations with any categorical variable can be observed.
Moreover, these groups are also associated with the vaginosis
variable.</p>
</div>
<div id="beta-diversity---nmds" class="section level2">
<h2>Beta Diversity - NMDS</h2>
<p>NMDS (Non-metric Multidimensional Scaling) analysis is a commonly
used non-metric ordination technique that helps visualize the similarity
or dissimilarity between samples based on a distance or similarity
matrix. In the context of microbiome data, this technique is
particularly useful for identifying patterns and significant changes in
microbiome composition among different samples or groups.</p>
<pre class="r"><code>mre &lt;- metar::virgo_nmds(mre, top_n = 50, save_files = FALSE)

metar::get_taxa(mre, &quot;virgo&quot;, &quot;nmds&quot;) %&gt;%
  purrr::pluck(&quot;ta1&quot;, &quot;top_50&quot;, &quot;all_adonis&quot;) %&gt;%
  dplyr::arrange(Pr..F.)</code></pre>
<pre><code>#&gt; # A tibble: 9 × 6
#&gt;   id                      Df SumOfSqs      R2      F Pr..F.
#&gt;   &lt;chr&gt;                &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;
#&gt; 1 pass_reads_quantile      3   16.5   0.200   18.9    0.001
#&gt; 2 vaginotype               2   38.0   0.459   96.9    0.001
#&gt; 3 vaginotype_condition     1   22.5   0.272   85.6    0.001
#&gt; 4 cluster                  2   37.9   0.459   96.7    0.001
#&gt; 5 Treatment_group          6    3.14  0.0379   1.47   0.022
#&gt; 6 Medical_center           1    0.606 0.00733  1.69   0.12 
#&gt; 7 Heme_test                1    0.531 0.00643  1.48   0.169
#&gt; 8 Arm                      1    0.446 0.00540  1.24   0.258
#&gt; 9 Timepoint                1    0.156 0.00188  0.432  0.873</code></pre>
<p>The Adonis test indicates that there are significant differences in
the taxonomic composition between the levels of the categorical
variables: <strong>pass_reads_quantile</strong>,
<strong>vaginotype</strong>, and <strong>vaginotype_condition</strong>,
but not on the categorical variable
<strong>Treatment_group</strong>.</p>
</div>
<div id="section-3" class="section level2 tabset">
<h2 class="tabset"></h2>
<pre class="r"><code>metar::get_cat(mre) %&gt;%
  dplyr::pull(1) %&gt;%
  purrr::walk( ~ {
    cat(&#39;\n\n### `&#39;, .x, &#39;`\n\n&#39;)
    metar::get_taxa(mre, &quot;virgo&quot;, &quot;nmds&quot;) %&gt;%
      purrr::pluck(&quot;ta1&quot;, &quot;top_50&quot;, &quot;categorical&quot;, .x) %&gt;%
      plot()
  })</code></pre>
<div id="treatment_group-3" class="section level3">
<h3><code>Treatment_group</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/nmds_groups-1.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="arm-3" class="section level3">
<h3><code>Arm</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/nmds_groups-2.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="heme_test-3" class="section level3">
<h3><code>Heme_test</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/nmds_groups-3.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="medical_center-3" class="section level3">
<h3><code>Medical_center</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/nmds_groups-4.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="pass_reads_quantile-3" class="section level3">
<h3><code>pass_reads_quantile</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/nmds_groups-5.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="vaginotype-3" class="section level3">
<h3><code>vaginotype</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/nmds_groups-6.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="vaginotype_condition-3" class="section level3">
<h3><code>vaginotype_condition</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/nmds_groups-7.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="section-4" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
<p>The nmds plots also indicate that there are significant differences
between the levels of the categorical variables:
<strong>pass_reads_quantile</strong>, <strong>vaginotype</strong>, and
<strong>vaginotype_condition</strong>, but not on the categorical
variable <strong>Treatment_group</strong>.</p>
</div>
<div id="differential-analyisis-ancom" class="section level2 tabset">
<h2 class="tabset">Differential Analyisis ANCOM</h2>
<pre class="r"><code># run ancom
mre &lt;- metar::virgo_ancom(mre, save_files = FALSE)

# plot ancom
metar::get_cat(mre) %&gt;%
  dplyr::pull(1) %&gt;%
  purrr::walk( ~ {
    plt &lt;- metar::get_taxa(mre, &quot;virgo&quot;, &quot;ancom&quot;) %&gt;%
      purrr::pluck(&quot;ta1&quot;, .x, &quot;plot&quot;)
    
    if (!is.null(plt)) {
      cat(&#39;\n\n### `&#39;, .x, &#39;`\n\n&#39;)
      plot(plt)
    }
  })</code></pre>
<div id="treatment_group-4" class="section level3">
<h3><code>Treatment_group</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/run_ancom-1.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="arm-4" class="section level3">
<h3><code>Arm</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/run_ancom-2.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="heme_test-4" class="section level3">
<h3><code>Heme_test</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/run_ancom-3.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="medical_center-4" class="section level3">
<h3><code>Medical_center</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/run_ancom-4.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="pass_reads_quantile-4" class="section level3">
<h3><code>pass_reads_quantile</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/run_ancom-5.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="vaginotype-4" class="section level3">
<h3><code>vaginotype</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/run_ancom-6.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="vaginotype_condition-4" class="section level3">
<h3><code>vaginotype_condition</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/run_ancom-7.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="section-5" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
<p>ANCOM is a statistical method designed specifically for the
compositional analysis of microbiome data. It’s used when you have
microbiome abundance data and want to identify features that are
differentially abundant across different groups or conditions.</p>
<p>W Statistic: In ANCOM, the “W statistic” is a measure used to assess
the significance of differential abundance for each feature (taxa or
OTUs). The W statistic measures how many times the abundance of a
particular feature is significantly different from the abundance
expected by chance across the groups being compared.</p>
<p>CLR (Centered Log-Ratio) Mean Difference: The CLR mean difference
represents the mean difference in the log-ratio transformed abundances
of a feature between two groups. The log-ratio transformation is applied
to the relative abundances of features to handle compositional data
properly. The mean difference indicates the average change in abundance
of a feature between the groups after applying the log-ratio
transformation.</p>
<p>In conclusion, when using ANCOM with a categorical variable, the W
statistic helps identify significant differences in feature abundance
across groups, while the CLR mean difference provides additional insight
into the direction and magnitude of these differences after applying the
log-ratio transformation.</p>
<p>Features with high W statistic values are considered significantly
differentially abundant between groups. The CLR mean difference helps to
understand the direction and magnitude of the difference in abundance
between groups after transformation.</p>
<p>Thus, these plots indicate us that there are abundance values that
are significantly different between
<strong>pass_reads_quantile</strong>, <strong>vaginotype</strong> and
<strong>vaginotype_condition</strong> groups, but not between
<strong>Treatment_group</strong> groups. Regarding pass_reads_quantile,
these differences are bigger when comparing c_4 group against others
(but do not occur on c_1 vs c_2 or c_1 vs c_3 and c_2 vs c_3), and
regarding vaginotype, these differences are bigger when comparing
<strong>L. Crispatus</strong> and <strong>G. Vaginalis</strong>
groups.</p>
</div>
<div id="differential-analyisis-lefse" class="section level2 tabset">
<h2 class="tabset">Differential Analyisis Lefse</h2>
<pre class="r"><code># run lefse
mre &lt;- metar::virgo_lefse(mre, save_files = FALSE)

# plot lefse
metar::get_cat(mre) %&gt;%
  dplyr::pull(1) %&gt;%
  purrr::walk(~ {
    plt &lt;- metar::get_taxa(mre, &quot;virgo&quot;, &quot;lefse&quot;) %&gt;%
      purrr::pluck(&quot;ta1&quot;, .x, &quot;plot&quot;)
    
    if (!is.null(plt)) {
      cat(&#39;\n\n### `&#39;, .x, &#39;`\n\n&#39;)
      plot(plt)
    }
  })</code></pre>
<div id="treatment_group-5" class="section level3">
<h3><code>Treatment_group</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/run_lefse-1.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="medical_center-5" class="section level3">
<h3><code>Medical_center</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/run_lefse-2.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="pass_reads_quantile-5" class="section level3">
<h3><code>pass_reads_quantile</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/run_lefse-3.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="vaginotype-5" class="section level3">
<h3><code>vaginotype</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/run_lefse-4.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
<div id="vaginotype_condition-5" class="section level3">
<h3><code>vaginotype_condition</code></h3>
<p><img src="figure/103_AIDSVax_effect.Rmd/run_lefse-5.png" width="1260" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="section-6" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
<p>LEfSe is a method used for identifying features that are
differentially abundant between two or more biological classes. It
combines the statistical rigor of linear discriminant analysis (LDA)
with the biological relevance of effect size estimation.</p>
<p>reorder(names, scores): In the context of LEfSe results, the
“reorder(names, scores)” refers to the reordering of the features based
on their scores. The scores represent the effect size of each feature in
discriminating between the biological classes. By reordering the
features based on their scores, you can visually prioritize the most
discriminative features.</p>
<p>LDA score (log10): The LDA score represents the effect size of a
feature in discriminating between the biological classes. It’s
calculated based on the combination of mean differences and standard
deviations of feature abundances between classes, as assessed by linear
discriminant analysis (LDA). The log10 transformation of the LDA score
is often used to make the distribution of scores more symmetric and
easier to interpret.</p>
<p>In conclusion, when using LEfSe results with a categorical variable,
reordering features based on their LDA scores and visualizing them using
log10-transformed scores can help prioritize and interpret the most
discriminative features associated with the biological classes.</p>
<p>Features with higher LDA scores (either positive or negative) are
considered more discriminative between the biological classes. By
sorting and visualizing the features based on their LDA scores, you can
identify the most biologically relevant and statistically significant
features associated with the categorical variable of interest.</p>
<p>If we consider features with LDA scores above 3 as biologically
relevant, these plots indicate us that <strong>Lactobacillus
Iners</strong>, <strong>Lactobacillus Crispatus</strong> and
<strong>Gardnerella Vaginalis</strong> are the species that explain the
difference between healthy and vaginosis groups.</p>
</div>
<div id="conclusions" class="section level2">
<h2>Conclusions</h2>
<p>After performing all these tests, we can conclude that there are no
differences between treatment groups or between study arms at
time-points 0, and 14. This means that there are no differences in terms
of microbiome composition between the application of ALVAC or in
combination with AIDSVax</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>#&gt; R version 4.1.2 (2021-11-01)
#&gt; Platform: x86_64-pc-linux-gnu (64-bit)
#&gt; Running under: Ubuntu 22.04.4 LTS
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0
#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0
#&gt; 
#&gt; locale:
#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#&gt;  [3] LC_TIME=es_ES.UTF-8        LC_COLLATE=en_US.UTF-8    
#&gt;  [5] LC_MONETARY=es_ES.UTF-8    LC_MESSAGES=en_US.UTF-8   
#&gt;  [7] LC_PAPER=es_ES.UTF-8       LC_NAME=C                 
#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
#&gt; [11] LC_MEASUREMENT=es_ES.UTF-8 LC_IDENTIFICATION=C       
#&gt; 
#&gt; attached base packages:
#&gt; [1] grid      stats     graphics  grDevices utils     datasets  methods  
#&gt; [8] base     
#&gt; 
#&gt; other attached packages:
#&gt; [1] ComplexHeatmap_2.18.0 ggplot2_3.5.0         magrittr_2.0.3       
#&gt; [4] workflowr_1.7.1      
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;   [1] tidyr_1.3.1                 knitr_1.45                 
#&gt;   [3] multcomp_1.4-25             DelayedArray_0.20.0        
#&gt;   [5] wesanderson_0.3.7           data.table_1.15.4          
#&gt;   [7] rpart_4.1.23                RCurl_1.98-1.14            
#&gt;   [9] doParallel_1.0.17           generics_0.1.3             
#&gt;  [11] BiocGenerics_0.40.0         TH.data_1.1-2              
#&gt;  [13] callr_3.7.6                 metar_0.1.5                
#&gt;  [15] future_1.33.1               correlation_0.8.4          
#&gt;  [17] tzdb_0.4.0                  bayesm_3.1-6               
#&gt;  [19] lubridate_1.9.3             httpuv_1.6.14              
#&gt;  [21] SummarizedExperiment_1.24.0 xfun_0.43                  
#&gt;  [23] hms_1.1.3                   jquerylib_0.1.4            
#&gt;  [25] evaluate_0.23               promises_1.2.1             
#&gt;  [27] DEoptimR_1.1-3              fansi_1.0.6                
#&gt;  [29] caTools_1.18.2              igraph_2.0.3               
#&gt;  [31] htmlwidgets_1.6.4           tensorA_0.36.2.1           
#&gt;  [33] kSamples_1.2-10             stats4_4.1.2               
#&gt;  [35] Rmpfr_0.9-5                 paletteer_1.6.0            
#&gt;  [37] purrr_1.0.2                 dplyr_1.1.4                
#&gt;  [39] survcomp_1.44.1             ggpubr_0.6.0               
#&gt;  [41] backports_1.4.1             insight_0.19.8             
#&gt;  [43] permute_0.9-7               prismatic_1.1.1            
#&gt;  [45] compositions_2.0-8          libcoin_1.0-10             
#&gt;  [47] MatrixGenerics_1.6.0        vctrs_0.6.5                
#&gt;  [49] Biobase_2.54.0              here_1.0.1                 
#&gt;  [51] ggmosaic_0.3.3              abind_1.4-5                
#&gt;  [53] cachem_1.0.8                withr_3.0.0                
#&gt;  [55] ggforce_0.4.2               robustbase_0.99-2          
#&gt;  [57] checkmate_2.3.1             vegan_2.6-4                
#&gt;  [59] cluster_2.1.6               ape_5.7-1                  
#&gt;  [61] lazyeval_0.2.2              crayon_1.5.2               
#&gt;  [63] pkgconfig_2.0.3             SuppDists_1.1-9.7          
#&gt;  [65] labeling_0.4.3              tweenr_2.0.3               
#&gt;  [67] GenomeInfoDb_1.30.1         nlme_3.1-164               
#&gt;  [69] statsExpressions_1.5.3      nnet_7.3-19                
#&gt;  [71] rlang_1.1.3                 globals_0.16.2             
#&gt;  [73] lifecycle_1.0.4             sandwich_3.1-0             
#&gt;  [75] survivalROC_1.0.3.1         phyloseq_1.38.0            
#&gt;  [77] rprojroot_2.0.4             polyclip_1.10-6            
#&gt;  [79] matrixStats_1.2.0           datawizard_0.9.1           
#&gt;  [81] Matrix_1.6-5                carData_3.0-5              
#&gt;  [83] zoo_1.8-12                  Rhdf5lib_1.16.0            
#&gt;  [85] boot_1.3-30                 base64enc_0.1-3            
#&gt;  [87] whisker_0.4.1               GlobalOptions_0.1.2        
#&gt;  [89] processx_3.8.4              png_0.1-8                  
#&gt;  [91] viridisLite_0.4.2           rjson_0.2.21               
#&gt;  [93] PMCMRplus_1.9.10            bootstrap_2019.6           
#&gt;  [95] parameters_0.21.5           bitops_1.0-7               
#&gt;  [97] getPass_0.2-4               KernSmooth_2.23-22         
#&gt;  [99] rhdf5filters_1.6.0          Biostrings_2.62.0          
#&gt; [101] shape_1.4.6.1               stringr_1.5.1              
#&gt; [103] multcompView_0.1-9          coin_1.4-3                 
#&gt; [105] parallelly_1.37.1           lefser_1.4.0               
#&gt; [107] readr_2.1.5                 rstatix_0.7.2              
#&gt; [109] S4Vectors_0.32.4            ggsignif_0.6.4             
#&gt; [111] rmeta_3.0                   scales_1.3.0               
#&gt; [113] memoise_2.0.1               plyr_1.8.9                 
#&gt; [115] gplots_3.1.3.1              zlibbioc_1.40.0            
#&gt; [117] compiler_4.1.2              RColorBrewer_1.1-3         
#&gt; [119] clue_0.3-65                 snakecase_0.11.1           
#&gt; [121] cli_3.6.2                   ade4_1.7-22                
#&gt; [123] XVector_0.34.0              listenv_0.9.1              
#&gt; [125] patchwork_1.2.0             ps_1.7.6                   
#&gt; [127] htmlTable_2.4.2             Formula_1.2-5              
#&gt; [129] MASS_7.3-60.0.1             mgcv_1.9-1                 
#&gt; [131] ggside_0.3.1                tidyselect_1.2.1           
#&gt; [133] stringi_1.8.3               forcats_1.0.0              
#&gt; [135] highr_0.10                  yaml_2.3.8                 
#&gt; [137] ANCOM_2.2                   ggrepel_0.9.5              
#&gt; [139] sass_0.4.9                  tools_4.1.2                
#&gt; [141] timechange_0.3.0            future.apply_1.11.1        
#&gt; [143] parallel_4.1.2              circlize_0.4.16            
#&gt; [145] rstudioapi_0.15.0           foreach_1.5.2              
#&gt; [147] foreign_0.8-86              git2r_0.33.0               
#&gt; [149] janitor_2.2.0               gridExtra_2.3              
#&gt; [151] prodlim_2023.08.28          farver_2.1.1               
#&gt; [153] digest_0.6.35               lava_1.8.0                 
#&gt; [155] BWStest_0.2.3               Rcpp_1.0.12                
#&gt; [157] GenomicRanges_1.46.1        car_3.1-2                  
#&gt; [159] broom_1.0.5                 performance_0.10.9         
#&gt; [161] later_1.3.2                 httr_1.4.7                 
#&gt; [163] effectsize_0.8.6            colorspace_2.1-0           
#&gt; [165] fs_1.6.3                    IRanges_2.28.0             
#&gt; [167] splines_4.1.2               rematch2_2.1.2             
#&gt; [169] multtest_2.50.0             plotly_4.10.4              
#&gt; [171] gmp_0.7-4                   jsonlite_1.8.8             
#&gt; [173] zeallot_0.1.0               modeltools_0.2-23          
#&gt; [175] R6_2.5.1                    Hmisc_5.1-1                
#&gt; [177] pillar_1.9.0                htmltools_0.5.8            
#&gt; [179] glue_1.7.0                  fastmap_1.1.1              
#&gt; [181] codetools_0.2-19            mvtnorm_1.2-4              
#&gt; [183] utf8_1.2.4                  lattice_0.22-5             
#&gt; [185] bslib_0.7.0                 tibble_3.2.1               
#&gt; [187] logger_0.3.0                gtools_3.9.5               
#&gt; [189] survival_3.5-8              rmarkdown_2.26             
#&gt; [191] biomformat_1.22.0           munsell_0.5.1              
#&gt; [193] GetoptLong_1.0.5            rhdf5_2.38.1               
#&gt; [195] GenomeInfoDbData_1.2.7      iterators_1.0.14           
#&gt; [197] ggstatsplot_0.12.2          reshape2_1.4.4             
#&gt; [199] gtable_0.3.4                bayestestR_0.13.2</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
