---
title: "CHIR HIV RV306 Vaginal - Global Description"
author: "Francesc Català-Moll & Oriol Careta Borràs"
output: html_document
output_dir: "`here::here('docs')`"
site: workflowr::wflow_site
mainfont: Helvetica
date: "`r format(Sys.time(), '%Y-%m-%d')`"
header-includes:
- \usepackage{fancyhdr}
- \usepackage{lipsum}
- \pagestyle{fancy}
- \fancyhead[CO,CE]{Flamingo - CIHR}
- \fancyfoot[LO,RE]{Oriol Careta Borras}
- \fancyhead[LE,RO]{\thepage}
- \fancyfoot[CO,CE]{}
- \usepackage[default]{sourcesanspro}
- \usepackage[T1]{fontenc}
- \usepackage{eso-pic,graphicx,transparent}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  cache = FALSE,
  autodep = TRUE,
  warning = FALSE,
  message = FALSE,
  dpi = 180,
  fig.align = "center",
  comment = "#>"
)
knitr::opts_knit$set(root.dir = here::here())
library(magrittr)
library(ggplot2)
```

## Introduction

We are interested in a general description of the dataset in terms of microbiome composition and function. We are using all available samples (481) from all patients (118). The dataset is contains vaginal microbiome data extracted 2 weeks after administering the vaccine regimen at the corresponding study time-points, following the RV306 trial:

![Sudy Design](assets/Study.jpg)

## Load mre

```{r, load_mre}
## mre init & filter
bucket <- "s3://cihr-hiv-rv306/vaginal/metagenome/WMGS"
out_dir <- here::here("data/preprocessed/")
dir.create(out_dir, showWarnings = FALSE, recursive = TRUE)

mre <- metar::metarExperiment_aws(bucket = bucket, wd = out_dir)

# Create new dataframe with the variable to add
df <- mre@metadata@metadata_df
new_df <- df %>% 
  dplyr::select(SampleID, Treatment_group) %>% 
  dplyr::mutate(
    Treatment_group_1 = dplyr::case_when(
      Treatment_group %in% "t_4" ~ "t_4A",
      Treatment_group %in% "t_no_boost" ~ "t_4A",
      TRUE ~ as.character (Treatment_group)
    )
  ) %>% 
  dplyr::select(SampleID, Treatment_group_1)

# Add the new var to the mre
mre <- metar::add_metadata_var(mre, new_df, id_col = "SampleID")

# Add the new var as categorical var
mre <- metar::add_cat(mre, "Treatment_group_1", "Set1")

mre <- 
  metar::get_meta(mre) %>% 
  dplyr::pull(SampleID) %>% 
  metar::filter_samples(mre, sample_ids = .)

mre
```

## QC

We will perform a brief QC as we have already performed it.

```{r absolute_counts, fig.width=15, fig.height=10}
## def vars
bucket <- "s3://cihr-hiv-rv306"
filt_data_obj <- "vaginal/metagenome/WMGS/RawData/DataFiltering_Boxplot.rds"

## load seq analysis data
df <- 
  aws.s3::s3readRDS(filt_data_obj, bucket) %>% 
  .[[1]] %>% ## Get data tibble from ggplot2 object
  tibble::as_tibble() %>% 
  dplyr::filter(SampleID %in% metar::get_meta(mre)$SampleID)

## Plot absolute
df %>%
  ggplot(aes(SampleID, value, fill = Step)) +
  geom_bar(position = "stack", stat = "identity", alpha = 0.8, colour = 0.0, width = 0.8) +
  scale_fill_brewer(palette = "Set1") +
  theme_minimal() +
  theme(axis.text.x = element_blank())
```

Here we can see the total number of pair-ended reads (PE reads) of each sample separated by QFail reads, QPass reads and Human reads. Samples appear ordered depending on the amount of Qpass reads. Thus, here we cannot only see the amount of QPass reads of each sample but also the sampling depth of each sample.

```{r relative_counts,fig.width=15, fig.height=10}
## Plot relative
df %>%
  ggplot(aes(SampleID, value, fill = Step)) +
  geom_bar(position = "fill", stat = "identity", alpha = 0.8, colour = 0.0) +
  scale_fill_brewer(palette = "Set1") +
  theme_minimal() +
  theme(axis.text.x = element_blank())
```

Here we can see the relative number of pair-ended reads (PE reads) of each sample separated by QFail reads, QPass reads and Human reads. Samples appear ordered depending on the total amount of Qpass reads. As it is relativized, we are not able to see sample depth but we can compare the percentage of QPass reads from the total amount on reads between all samples.

We will also add a categorical variable where we distribute the QPass reads in quantiles, as usually a higher number of QPass reads means higher diversity observed, but we want to check if it is be the case.

```{r new_metadata}
## Add new metadata var
mre <- 
  dplyr::filter(df, Step == "QPassReads") %>% 
  dplyr::mutate(
    pass_reads_quantile = findInterval(value, quantile(value)),
    pass_reads_quantile = dplyr::case_when(
      pass_reads_quantile == 5 ~ "c_4", 
      TRUE ~ stringr::str_c("c_", pass_reads_quantile)
    )
  ) %>% 
  dplyr::select(SampleID, pass_reads_quantile) %>% 
  metar::add_metadata_var(mre, .) %>% 
  metar::add_cat("pass_reads_quantile", "Pastel2")
```

## {-}

## Alpha Diversity - Rarefaction {.tabset}

Note that diversity analysis here are delicate to perform due to the limitations in sampling.

From the taxonomic composition we'll derive some alpha-diversity ecological indices and compare them among groups.

We also want to use the VIRGO catalog mapping-derived gene richness. However, this can be problematic for gene richness since the initial reads per sample is very low in some cases. This low numbers may be enough to describe the taxonomic composition, specially if the microbiome is "simple", but may fail to capture richness/diversity.

```{r alpha_d_rarefraction, results='asis'}
## run rarefaction
mre <- metar::virgo_rarefaction(mre, save_files = FALSE)

metar::get_cat(mre) %>%
  dplyr::pull(1) %>%
  .[-1] %>% # Remove "Treatment_group"
  purrr::walk(~ {
    cat('\n\n### `', .x, '`\n\n')
    metar::get_diversity(mre, type = "virgo", res_slot = "rarefaction") %>%
      purrr::pluck("categorical", .x, "rarefaction") %>%
      plot()
  })
```

## {-}

The rarefaction curve shows how far we are from assymptotically saturating richness signal from this catalog at the sampling depth we have (after filtering human sequences). Thus, we need to be aware of the limitations of gene richness analysis and interpret with caution. When comparing between variable groups, sampling depth seem to be no different between them, except, logically, for the variable Qpass reads quantile, where, depending on the quantile, they have more or less sample depth.

## Alpha Diversity - Boxplots {.tabset }

```{r alpha_d_boxplots, results='asis'}
## plot gene richness
metar::get_cat(mre) %>%
  dplyr::pull(1) %>%
  .[-1] %>% # Remove "Treatment_group"
  purrr::walk(~ {
    cat('\n\n### `', .x, '`\n\n')
    metar::get_diversity(mre, type = "virgo", res_slot = "rarefaction") %>%
      purrr::pluck("categorical", .x, "boxplots", "GeneNumber") %>%
      plot()
  })
```

## {-}

The gene richness boxplots indicate that there are no significant differences between groups of variables in terms of gene richness, except when separating by treatment group, where the gene richness of t_2 is significantly higher compared to t_4A. There are also significant differences between gene richness when observing by Qpass reads quantiles, as a higher number of Qpass reads (higher sampling depth) usually leads to a higher gene richness.

## Alpha Diversity - Longitudinal {.tabset }

To further explore the aforementioned observations, we will verify whether gene diversity varies across time-points for the categorical variables.

```{r lon_gene_richness, results='asis'}
metar::get_cat(mre) %>%
  dplyr::pull(1) %>%
  .[-1] %>% # Remove "Treatment_group"
  purrr::walk(~ { 
    cat('\n\n### `', .x, '`\n\n')
    metar::get_diversity(mre, type = "virgo", res_slot = "rarefaction") %>%
      purrr::pluck("longitudinal_rel", "categorical", .x) %>%
      plot()
  })
```

## {-}

In these graphs, we can see the changes in the relative gene richness over time of samples separated by categorical variable groups. It seems that while gene richness is stable on the active arm group over time, gene richness on the placebo group is more variable. Regarding the other variables, gene richness seems constant over time.

## Beta-Diversity - Species Composition

Let's visualize composition at the species level using barplots.

```{r abundance_barplots, fig.width=20, fig.height=10}
## run barplots 
mre <- metar::virgo_barplots(mre, top_n = 50, save_files = FALSE)

## Plot virgo barplots
mre %>%
  metar::get_taxa("virgo", "barplots") %>%
  purrr::pluck("ta1", "top_50", "rel_abundance", "bray_hclust_order_barplot")
```

The relative abundance barplots in hierarchical clustering order depicting taxonomic diversity per sample reveal the presence of three distinct groups:

* The first group is primarily characterized by the dominance of **Gardnerella vaginalis**, occasionally accompanied by co-dominance with **Atopobium vaginae**. Notably, this group exhibits a higher level of taxonomic diversity compared to the other two groups. Both **Gardnerella vaginalis**, a gram-negative facultative anaerobic bacterium, and **Atopobium vaginae**, a gram-positive strict anaerobic bacterium, have been associated with bacterial vaginosis (BV) and its characteristic symptoms.
* The second group, is characterized by a strong dominance of **Lactobacillus iners**. **Lactobacillus iners** is a common bacterium found in the vaginal microbiota, playing a vital role in maintaining vaginal health. It helps to maintain an acidic pH, creating an unfavorable environment for harmful microorganisms. **Lactobacillus iners** contributes to the stability and balance of the vaginal microbiota, promoting overall vaginal health and potentially protecting against infections like bacterial vaginosis.
* The third group is mainly characterized by a dominant presence of **Lactobacillus crispatus**, although some samples in this group also show high levels of **Lactobacillus iners**. **Lactobacillus crispatus** is a beneficial bacterium commonly found in the vaginal microbiota. It helps maintain a healthy vaginal environment by producing lactic acid and supporting acidity.

```{r pred_taxa}
## Count of two predominant taxas per sample
mre %>% 
  metar::get_taxa("virgo", "barplots") %>%
  purrr::pluck("ta1", "top_50", "rel_abundance", "bray_hclust_order_barplot", "data") %>%
  dplyr::group_by(SampleID) %>%
  dplyr::slice_max(Abundance, n = 2) %>%
  dplyr::ungroup() %>%
  dplyr::count(Species, sort = TRUE)
```

To confirm these observations, we created a table where we counted the two most dominant species per each sample. It can be seen that the most abundant species are **Gardnerella vaginalis**, **Lactobacillus iners** and **Lactobacillus crispatus**, followed by **Atopobium vaginae**, which on the abundance barplot was found in codominance with **Gardnerella vaginalis** on some samples. The fourth and fifth more common species are also **Lactobacillus**.

## Beta-Diversity - Cluestering

In the previous barplots, the x-axis is defined according to ward.D2, bray-based clustering. We'll now transform abundances and project them into a heatmap, to be able to visualize association with variables of interest.

```{r clust_heatmap, fig.height=10}
mre <- metar::virgo_heatmap(mre, top_n = 50, save_files = FALSE)

mre %>% 
  metar::get_taxa("virgo", "heatmaps") %>%
  purrr::pluck("ta1", "top_50") %>%
  ComplexHeatmap::draw(heatmap_legend_side = "left", annotation_legend_side = "bottom")
```

In this heatmap with hierarchical clustering, we can clearly see the three groups, but there is no apparent association between composition and metadata variables. In some samples dominated by **Gardnerella vaginalis**, there is co-dominance with **Atopobium vaginae** present.

## Beta-Diversity - Unsupervised Cluestering

We have seen that the vaginal microbiome composition is quite discretized in many of the samples. That is, there is a single species which dominate in most of the samples, while the rest have a combination fo the rest of the species, with higher dominance indices (we can check that separately). So, the question is whether there are composition types (from now on, called vaginotypes) that we can detect.

For consistence with the rest of the analysis, we'll perform the PAM (Partitioning Around Medoids) clustering analysis using all samples in order to broadly detect these vagintoypes and be able to uniquely assign each sample to a vaginotype, while ensuring that this assignment does not changes in downstream analysis.

The PAM algorithm searches for k representative objects in a data set (k medoids) and then assigns each object to the closest medoid in order to create clusters. Its aim is to minimize the sum of dissimilarities between the objects in a cluster and the center of the same cluster (medoid).

```{r clusters}
## run nmds
mre <- metar::virgo_nmds(mre, top_n = 50, save_files = FALSE)

## Silhouette plot
mre %>% 
  metar::get_taxa("virgo", "nmds") %>%
  purrr::pluck("ta1", "top_50", "clustering_results", "silhouette_plt")
```

The Silhouette plot shows that the silhouette coefficient is highest when k=3, suggesting that’s the optimal number of clusters. This may be pointing to 3 different vaginotypes in the entire sample set. Moreover, it agrees with what was observed in the barplot and the heatmap.

```{r cluster_sum}
## Cluster summary
mre %>% 
  metar::get_taxa("virgo", "nmds") %>%
  purrr::pluck("ta1", "top_50", "clustering_results", "sample_cluster_labels") %>%
  dplyr::count(cluster)
```

This table shows us the number of samples that are classified into each of these 3 clusters.

Nevertheless, we are interested in the main characteristics of these vaginotypes, so we will use the cluster label to plot a heatmap and visualize main characteristics. 

```{r cluster_nmds}
## Cluster nmds
mre %>% 
  metar::get_taxa("virgo", "nmds") %>%
  purrr::pluck("ta1", "top_50", "categorical", "cluster")
```

Apparently:

* Cluster 1: Is enriched with **L. crispatus**, although some samples in this group also show high levels of **L. iners** and other Lactobacillus.

* Cluster 2: Is mainly **G. vaginalis**, occasionally accompanied by co-dominance with **A. vaginae**.

* Cluster 3: Is dominated by **L. iners**.

We will first create a variable for these clusters, and we also added a categorical variable based on whether the vaginotype has pathological potential or not. We call the categorical variable **vaginotype condition** which includes the categories **healty** and **vaginosis**.

```{r vaginotypes_as_var}
## Prepare var
vaginotypes <- 
  mre %>% 
  metar::get_taxa("virgo", "nmds") %>% 
  purrr::pluck("ta1", "top_50", "clustering_results", "sample_cluster_labels") %>% 
  dplyr::mutate(
    vaginotype = dplyr::case_when(
      cluster == 1 ~ "c1_L_crispatus", 
      cluster == 2 ~ "c2_G_vaginalis", 
      cluster == 3 ~ "c3_L_iners"
    ), 
    vaginotype_condition = dplyr::if_else(cluster == 2, "vaginosis", "no_vaginosis")
  ) %>% 
  dplyr::select(SampleID, vaginotype, vaginotype_condition)

## Update mre metadata
mre <- 
  metar::add_metadata_var(mre, vaginotypes) %>% 
  metar::add_cat(var = "vaginotype", palette = "Accent") %>% 
  metar::add_cat(var = "vaginotype_condition", palette = "Set1")

## Save mre with vaginotype metadata variable
readr::write_rds(mre, here::here("data", "preprocessed", "mre_vaginotype.rds"))
```

If we take a closer look to clusters, in terms of Alpha Diversity:

## Alpha Diversity - Rarefraction {.tabset }

```{r, alpha_d_rarefraction_2, results='asis'}
## Run rarefaction for new var
mre <- metar::virgo_rarefaction(mre, save_files = FALSE)

metar::get_cat(mre) %>%
  dplyr::pull(1) %>%
  .[-1] %>% # Remove "Treatment_group"
  purrr::walk(~ {
    cat('\n\n### `', .x, '`\n\n')
    metar::get_diversity(mre, type = "virgo", res_slot = "rarefaction") %>%
      purrr::pluck("categorical", .x, "rarefaction") %>%
      plot()
  })
```

## {-}

The rarefaction curve shows that when adding the categorical variable vaginotype, the samples which are more rich in **G. Vaginalis** are closer to asymptotically saturating richness signal from this catalog, mainly because they tend to have a higher sampling depth.

## Alpha Diversity - Boxplots {.tabset }

```{r alpha_d_boxplots_2}
## plot gene richness
metar::get_cat(mre) %>%
  dplyr::pull(1) %>%
  .[-1] %>% # Remove "Treatment_group"
  purrr::walk(~ {
    cat('\n\n### `', .x, '`\n\n')
    metar::get_diversity(mre, type = "virgo", res_slot = "rarefaction") %>%
      purrr::pluck("categorical", .x, "boxplots", "GeneNumber") %>%
      plot()
  })
```

## {-}

After representing the gene richness for each level of each categorical variable, we observed that there are also significant differences within the categorical variables vaginotype. We noticed that Cluster 2 (dominated by **G. Vaginalis** and potentially associated with vaginosis) exhibits higher gene diversity compared to the other two clusters. Furthermore, it is worth mentioning that the other two clusters also display variations in gene diversity among themselves.

## Alpha Diversity - Longitudinal {.tabset }

To further explore the aforementioned observations, we will verify whether gene diversity varies across timepoints for the categorical variables.

```{r lon_gene_richness_2, results='asis'}
metar::get_cat(mre) %>%
  dplyr::pull(1) %>%
  .[-1] %>% # Remove "Treatment_group"
  purrr::walk(~ { 
    cat('\n\n### `', .x, '`\n\n')
    metar::get_diversity(mre, type = "virgo", res_slot = "rarefaction") %>%
      purrr::pluck("longitudinal_rel", "categorical", .x) %>%
      plot()
  })
```

## {-}

In these graphs, we can see the changes in the relative gene richness over time of samples separated by categorical variable groups. It seems that the cluster dominated by **L. Iners** has a less stable microbiome over time.


## Beta-Diversity - Cluestering Heatmap

In order to visualize how the newly generated categorical variables cluster, we will once again represent the Heatmap for the 50 most abundant bacterial species.

```{r hierarchical_clustering_new_vars, fig.height=10}
## run heatmap
mre <- metar::virgo_heatmap(mre, top_n = 50, save_files = FALSE)

## plot heatmap
mre %>% 
  metar::get_taxa("virgo", "heatmaps") %>%
  purrr::pluck("ta1", "top_50") %>%
  ComplexHeatmap::draw(heatmap_legend_side = "left", annotation_legend_side = "bottom")
```

In this heatmap with hierarchical clustering, we can clearly see that the three bacterial groups correspond to the defined clusters. Moreover, we can clearly see the association between the variables vaginotype and vaginotype_condition and species composition.

## {-}

## Alpha Diversity - NMDS 

NMDS (Non-metric Multidimensional Scaling) analysis is a commonly used non-metric ordination technique that helps visualize the similarity or dissimilarity between samples based on a distance or similarity matrix. In the context of microbiome data, this technique is particularly useful for identifying patterns and significant changes in microbiome composition among different samples or groups.

```{r adonis_test}
mre <- metar::virgo_nmds(mre, top_n = 50, save_files = FALSE)
metar::get_taxa(mre, "virgo", "nmds") %>%
  purrr::pluck("ta1", "top_50", "all_adonis") %>%
  dplyr::arrange(Pr..F.)
```

The Adonis test indicates that there are significant differences in the taxonomic composition between the groups of the categorical variables: **Treatment_group_1**, **pass_reads_quantile**, **vaginotype**, and **vaginotype_condition**.

## {.tabset }

```{r nmds_groups, results='asis'}
metar::get_cat(mre) %>%
  dplyr::pull(1) %>%
  .[-1] %>% # Remove "Treatment_group"
  purrr::walk( ~ {
    cat('\n\n### `', .x, '`\n\n')
    metar::get_taxa(mre, "virgo", "nmds") %>%
      purrr::pluck("ta1", "top_50", "categorical", .x) %>%
      plot()
  })
```

## {-}

The nmds plots also indicate that there are significant differences between the levels of the categorical variables: **pass_reads_quantile**, **vaginotype**, and **vaginotype_condition**. Moreover, the species contributing to cluster differentiation can be seen. While **G. Vaginalis** and **A. Vaginae** contribute to the formation of cluster 2 (and to the appearance of vaginosis), **L. Crispatus** and **L. Iners** contribute to their respective clusters.

## Longitudinal Stability of Vaginotype within Individual Patients {.tabset}

At this point, it is clear that the categorical variable "vaginotype" is a significant component of the observed compositional variability. To further investigate this phenotypic characteristic, we ask the question of whether the vaginotype remains stable longitudinally within the same patient.

First, we generated alluvial plots for each categorical variable.

```{r alluvial_plot, results='asis'}
library(ggalluvial)
metar::get_cat(mre) %>%
  dplyr::pull(1) %>%
  .[-1] %>% # Remove "Treatment_group"
  purrr::walk(~ {
    cat('\n\n### `', .x, '`\n\n')
    p <- 
      metar::get_meta(mre) %>%
      dplyr::select(PatientID, !!dplyr::sym(.x), Timepoint) %>%
      ggplot(
        aes(
          as.factor(Timepoint),
          stratum = !!dplyr::sym(.x),
          alluvium = PatientID,
          fill = !!dplyr::sym(.x),
          label = !!dplyr::sym(.x)
        )
      ) +
      scale_x_discrete(expand = c(.1, 0)) +
      geom_flow(width = 1 / 4) +
      geom_stratum(alpha = .5, width = 1 / 4) +
      geom_text(stat = "stratum", size = 4) +
      theme_minimal() +
      theme(legend.position = "none")
    
    plot(p)
  })
```

## {-}

We can observe that the vaginotype variable does not remain stable over time. Some patients start with a potential vaginosis phenotype and then transition to a healthy phenotype, while others transition from a healthy phenotype to vaginosis. Additionally, some patients go through both stages cyclically.
