---
title: "CHIR HIV RV306 Vaginal - Groups Comparison"
author: "Oriol Careta BorrÃ s"
output: html_document
output_dir: "`here::here('docs')`"
site: workflowr::wflow_site
mainfont: Helvetica
date: "`r format(Sys.time(), '%Y-%m-%d')`"
header-includes:
- \usepackage{fancyhdr}
- \usepackage{lipsum}
- \pagestyle{fancy}
- \fancyhead[CO,CE]{Flamingo - CIHR}
- \fancyfoot[LO,RE]{Oriol Careta Borras}
- \fancyhead[LE,RO]{\thepage}
- \fancyfoot[CO,CE]{}
- \usepackage[default]{sourcesanspro}
- \usepackage[T1]{fontenc}
- \usepackage{eso-pic,graphicx,transparent}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  cache = FALSE,
  autodep = TRUE,
  warning = FALSE,
  message = FALSE,
  dpi = 180,
  fig.align = "center",
  comment = "#>"
)
knitr::opts_knit$set(root.dir = here::here())
library(magrittr)
library(ggplot2)
```

## Introduction

We are interested in comparing samples at two weeks after the last boost in terms of microbiome composition and function to answer the question are there differences in rectal microbiome composition between different vaccine treatments?. 

## MRE initialization and filter baseline samples

```{r load_data}
mre <- readr::read_rds(here::here("data", "preprocessed", "mre_vaginotype.rds"))

## mre init & filter
mre <- 
  metar::get_meta(mre) %>%
  dplyr::filter(Timepoint %in% c(50) & Treatment_group %in% c("t_1", "t_2", "t_3") | Timepoint %in% c(62) & Treatment_group %in% c("t_4A") | Timepoint %in% c(74) &   Treatment_group %in% c("t_4B")) %>%
  dplyr::pull(SampleID) %>%
  metar::filter_samples(mre, sample_ids = .)

mre
```

There are 107 samples from 107 patients which were collected at time-points 50, 62 and 74, depending on the group.

## Gene richness - Rarefaction {.tabset}

Note that diversity analysis here are delicate to perform due to the limitations in sampling.

From the taxonomic composition we'll derive some alpha-diversity ecological indices and compare them among groups.

We also want to use the VIRGO catalog mapping-derived gene richness. However, this can be problematic for gene richness since the initial reads per sample is very low in some cases. This low numbers may be enough to describe the taxonomic composition, specially if the microbiome is "simple", but may fail to capture richness/diversity.

```{r rarefraction_plots, results='asis'}
## run rarefaction
mre <- metar::virgo_rarefaction(mre, save_files = FALSE)

## plot rarefaction
metar::get_cat(mre) %>%
  dplyr::pull(1) %>%
  .[-1] %>% # Remove "Treatment_group"
  purrr::walk(~ {
    cat('\n\n### `', .x, '`\n\n')
    metar::get_diversity(mre, type = "virgo", res_slot = "rarefaction") %>%
      purrr::pluck("categorical", .x, "rarefaction") %>%
      plot()
  })
```

## {-}

The rarefraction curves (used to present relationship between gene richness and sequencing depth) show no apparent difference between groups in terms of diversity. Of note, rarefaction curve shows how far we are from asymptotically saturating richness signal from this catalog at the sampling depth we have (after filtering human sequences).  Thus, we need to be aware of the limitations of gene richness analysis and interpret with caution. It is worth mentioning that samples enriched in **G. Vaginalis** (and thus, with a vaginosis profile) seem to have a higher sample depth and gene richness compared to the other ones.

## Gene richness - Boxplots {.tabset }

```{r gene_richness, results='asis'}
## plot gene richness
metar::get_cat(mre) %>%
  dplyr::pull(1) %>%
  .[-1] %>% # Remove "Treatment_group"
  purrr::walk(~ {
    cat('\n\n### `', .x, '`\n\n')
    metar::get_diversity(mre, type = "virgo", res_slot = "rarefaction") %>%
      purrr::pluck("categorical", .x, "boxplots", "GeneNumber") %>%
      plot()
  })
```

## {-}

The gene richness boxplots indicate that there are no significant differences between groups in terms of gene richness, except when separating by the number of QPass reads by quantiles, where the gene richness on the 4th quantile group is significantly higher compared to 2nd quantile . This could be explained due to the fact that the sample depth is higher on this group which favors gene richness. There are also significant differences between vaginotype groups, but not between treatment groups.

## PCoA based on Bray Curtis 

```{r PCoA, fig.width=20, fig.height=5}
ps <- metar::get_phyloseq(mre, type = "virgo")
ps_RA <- phyloseq::transform_sample_counts(ps, function(x) x / sum(x))
bray_pcoa  <-
  phyloseq::ordinate(physeq = ps_RA, method = "PCoA", distance = "bray")

create_pcoa_plot <- function(ordination_obj, title, a, b) {
  plot <- phyloseq::plot_ordination(
    physeq = ps_RA,
    ordination = ordination_obj,
    color = "vaginotype",
    shape = "Arm",
    axes = c(a,b)
  ) +
  geom_point(size = 2.5) +
  scale_color_manual(values = c("deeppink4", "coral", "cadetblue4")) +
  scale_shape_manual(values = c(16, 2)) +
  theme_light() +
  ggtitle(title)
  
  return(plot)
}

pcoa_plot_PCo1_PCo2 <- create_pcoa_plot(bray_pcoa, "PCo1 vs PCo2", 1, 2)
pcoa_plot_PCo1_PCo3 <- create_pcoa_plot(bray_pcoa, "PCo1 vs PCo3", 1, 3)
pcoa_plot_PCo2_PCo3 <- create_pcoa_plot(bray_pcoa, "PCo2 vs PCo3", 2, 3)

gridExtra::grid.arrange(pcoa_plot_PCo1_PCo2, pcoa_plot_PCo1_PCo3, pcoa_plot_PCo2_PCo3, nrow = 1)
```

In the PCoA plots, it can be seen that the compositional variables that affect each cluster. PCo 1 and 2 explain the 85.2% of the variability (46.2% and 39% respectively), while PCo3 only explains the 4.5% of the variability. From the plots we can see that PCo1 is the one that explains the separation of cluster 2 (G. Vaginalis) from cluster 1 (L. Crispatus) and cluster 3 (L. Iners), while PCo2 explains the separation of cluster 1 from cluster 2 and 3. On the other hand, PCo3 is not useful to discriminate between clusters.

## Beta-Diversity - Species Composition

Let's visualize composition at the species level using barplots.

```{r top50_abundance_barplot, fig.width=20, fig.height=10}
## run barplots 
mre <- metar::virgo_barplots(mre, top_n = 50, save_files = FALSE)

## Plot virgo barplots
mre %>% 
  metar::get_taxa("virgo", "barplots") %>%
  purrr::pluck("ta1", "top_50", "rel_abundance", "bray_NMDS1_order_barplot") +
  theme(axis.text.x = element_blank())
```

 The clustering barplots depicting taxonomic diversity per sample reveal the presence of three distinct groups:

* The first group is primarily characterized by the dominance of **Gardnerella vaginalis**, occasionally accompanied by co-dominance with **Atopobium vaginae**. Notably, this group exhibits a higher level of taxonomic diversity compared to the other two groups. Both **Gardnerella vaginalis**, a gram-negative facultative anaerobic bacterium, and **Atopobium vaginae**, a gram-positive strict anaerobic bacterium, have been associated with bacterial vaginosis (BV) and its characteristic symptoms.
* The second group, is characterized by a strong dominance of **Lactobacillus iners**. **Lactobacillus iners** is a common bacterium found in the vaginal microbiota, playing a vital role in maintaining vaginal health. It helps to maintain an acidic pH, creating an unfavorable environment for harmful microorganisms. **Lactobacillus iners** contributes to the stability and balance of the vaginal microbiota, promoting overall vaginal health and potentially protecting against infections like bacterial vaginosis.
* The third group is mainly characterized by a dominant presence of **Lactobacillus crispatus**, although some samples in this group also show high levels of **Lactobacillus jensenii**. **Lactobacillus crispatus** is a beneficial bacterium commonly found in the vaginal microbiota. It helps maintain a healthy vaginal environment by producing lactic acid and supporting acidity. **Lactobacillus jensenii** is a gram-positive bacterium commonly found in the vaginal microbiota, particularly in women of reproductive age. It contributes to maintaining vaginal health by producing lactic acid and creating an acidic environment, which helps inhibit the growth of harmful bacteria. **Lactobacillus jensenii** is known for its ability to adhere to vaginal epithelial cells, which may enhance its protective effects against infections and maintain the balance of the vaginal microbiota.

```{r pred_taxa}
## Count of two predominant taxas per sample
mre %>% 
  metar::get_taxa("virgo", "barplots") %>%
  purrr::pluck("ta1", "top_50", "rel_abundance", "bray_NMDS1_order_barplot", "data") %>%
  dplyr::group_by(SampleID) %>%
  dplyr::slice_max(Abundance, n = 2) %>%
  dplyr::ungroup() %>%
  dplyr::count(Species, sort = TRUE)
```

In this table, we can see the to most dominant species of each sample summed up. It can be seen that the most abundant species are **Gardnerella vaginalis**, **Lactobacillus iners** and **Lactobacillus crispatus**, followed by **Atopobium vaginae**, which on the abundance barplot was found in codominance with **Gardnerella vaginalis** on some samples.

## Beta-Diversity - Cluestering

In the previous barplots, the x-axis is defined according to ward.D2, bray-based clustering. We'll now transform abundances and project them into a heatmap, to be able to visualize association with variables of interes.

```{r hierarchical_clustering, fig.height=10}
## run heatmap
mre <- metar::virgo_heatmap(mre, top_n = 50, save_files = FALSE)

## plot heatmap
library(ComplexHeatmap)
mre %>%
  metar::get_taxa("virgo", "heatmaps") %>%
  purrr::pluck("ta1", "top_50") %>%
  ComplexHeatmap::draw(heatmap_legend_side = "left", annotation_legend_side = "bottom")
```

In this hierarchical clustering heatmap, we can clearly see the three groups. It can be seen that the vaginotype **G. Vaginalis** is enriched in this specie, while the other two are dominated by **L. Crispatus** and **L. Iners**. Moreover, the **G. vaginalis** dominated group is highly polimicrobial.

## Beta Diversity - NMDS 

NMDS (Non-metric Multidimensional Scaling) analysis is a commonly used non-metric ordination technique that helps visualize the similarity or dissimilarity between samples based on a distance or similarity matrix. In the context of microbiome data, this technique is particularly useful for identifying patterns and significant changes in microbiome composition among different samples or groups.

```{r adonis_table}
mre <- metar::virgo_nmds(mre, top_n = 50, save_files = FALSE)

metar::get_taxa(mre, "virgo", "nmds") %>%
  purrr::pluck("ta1", "top_50", "all_adonis") %>%
  dplyr::arrange(Pr..F.)
```

The Adonis test indicates that there are significant differences in the taxonomic composition between the levels of the categorical variables: **pass_reads_quantile**, **vaginotype**, **vaginotype_condition** and **Heme_test**.

## {.tabset }

```{r nmds_groups, results='asis'}
metar::get_cat(mre) %>%
  dplyr::pull(1) %>%
  .[-1] %>% # Remove "Treatment_group"
  purrr::walk( ~ {
    cat('\n\n### `', .x, '`\n\n')
    metar::get_taxa(mre, "virgo", "nmds") %>%
      purrr::pluck("ta1", "top_50", "categorical", .x) %>%
      plot()
  })
```

## {-}

The nmds plots also indicate that there are significant differences between the levels of the categorical variables: **pass_reads_quantile**, **vaginotype**, and **vaginotype_condition**, but not on the categorical variable **Heme_test**

## Differential Analyisis ANCOM {.tabset }

```{r run_ancom, results='asis'}
# run ancom
mre <- metar::virgo_ancom(mre, save_files = FALSE)

# plot ancom
metar::get_cat(mre) %>%
  dplyr::pull(1) %>%
  .[-1] %>% # Remove "Treatment_group"
  purrr::walk( ~ {
    plt <- metar::get_taxa(mre, "virgo", "ancom") %>%
      purrr::pluck("ta1", .x, "plot")
    
    if (!is.null(plt)) {
      cat('\n\n### `', .x, '`\n\n')
      plot(plt)
    }
  })
```

## {-}

ANCOM plots indicate us that there are abundance values that are significantly different between **pass_reads_quantile**, **vaginotype** and **vaginotype_condition** groups, but not between **Treatment_group** groups. Regarding pass_reads_quantile, these differences are bigger when comparing c_4 group against others , and regarding vaginotype, these differences are bigger when comparing **L. Crispatus** and **G. Vaginalis** groups.

## Differential Analyisis Lefse {.tabset }

```{r run_lefse, results='asis'}
# run lefse
mre <- metar::virgo_lefse(mre, save_files = FALSE)

# plot lefse
metar::get_cat(mre) %>%
  dplyr::pull(1) %>%
  .[-1] %>% # Remove "Treatment_group"
  purrr::walk(~ {
    plt <- metar::get_taxa(mre, "virgo", "lefse") %>%
      purrr::pluck("ta1", .x, "plot")
    
    if (!is.null(plt)) {
      cat('\n\n### `', .x, '`\n\n')
      plot(plt)
    }
  })
```

## {-}

## Conclusions

After performing all these tests, we can conclude that there are no differences between treatment groups or between study arms at time-points 50, 62 and 74. This means that there are no differences between vaccination regimes.
